---
title: "DIAN Brain Age Analyses"
output: html_notebook
---

Clear workspace
```{r}
rm(list=ls())
```

#Load data and setup
```{r}
#load packages
pacman::p_load(tidyverse, corrplot, psych, data.table, pander, clean, Metrics, ggpubr, gtsummary, neuroCombat, longCombat, Metrics, DescTools, viridis, dutchmasters, ggsci, lme4, nlme, lmerTest, multilevelTools, splines, AICcmodavg, effects, mgcv, gamm4, ggthemes, effectsize, ez, heplots, irr, gridExtra, mediation)


get_legend<-function(myggplot){
  tmp <- ggplot_gtable(ggplot_build(myggplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}

#set path
################
# LOCAL PATH
################
base_path = "W:"

################
# SERVER PATH
################
#base_path = "data/nil-bluearc/ances/millarp"

#define df paths, output paths, and version date
df_path = file.path(base_path, "data/DIAN/DIAN_DF15_V1/")
minidf_path = file.path(base_path, "data/DIAN/DIAN_MiniDF/")

out_path = file.path(base_path, "dian_analyses/bag/results/")

ms_path = file.path("C:/Users/millarp/Box/Manuscripts/BPA3 DIAN DBN/BPA3 DIAN DBN Figures")

ver_date = "231011"

#load raw data from df files
raw_dbn = fread(file.path(base_path, "DeepBrainNet/Output_DIAN_df15/predictions.csv")) %>%
  separate(ID, c('imagid', 'visit', 'source', 'accession', 'freesurfer', 'fs_date'))
raw_dbn = subset(raw_dbn, visit != '')

#raw_demo = fread(file.path(df_path, "DEMOGRAPHICS_D1205.csv"))
raw_demo = fread(file.path(minidf_path, "DEMOGRAPHICS_D2219.csv")) %>%
  mutate(imagid = as.factor(IMAGID))

###NOT LOADING ALL ROWS FOR CLIN???
#raw_clin = fread(file.path(df_path, "CLINICAL_D1205_SHORT.csv"))
raw_clin = fread(file.path(minidf_path, "CLINICAL_D2219.csv")) %>%
  mutate(imagid = as.factor(IMAGID))

#raw_gen = fread(file.path(df_path, "GENETIC_D1205.csv"))
raw_gen = fread(file.path(minidf_path, "GENETIC_D2219.csv")) %>%
  mutate(imagid = as.factor(IMAGID))

raw_mri = fread(file.path(df_path, "IMAGING_dian_pj2_ADADCortSig.csv"))

#raw_lumi = fread(file.path(df_path, "lumi_D1205.csv"))
raw_lumi = fread(file.path(minidf_path, "lumipulse_mini_df_newid16.csv")) %>%
  mutate(LUMIPULSE_CSF_AB40 = as.numeric(LUMIPULSE_CSF_AB40))%>%
  mutate(LUMIPULSE_CSF_AB42 = as.numeric(LUMIPULSE_CSF_AB42))%>%
  mutate(LUMIPULSE_CSF_tTau = as.numeric(LUMIPULSE_CSF_tTau))%>%
  mutate(LUMIPULSE_CSF_pTau = as.numeric(LUMIPULSE_CSF_pTau))

raw_pib = fread(file.path(df_path, "pib_D1205.csv"))

#raw_cog = fread(file.path(df_path, "psychometric_D1205.csv"))
raw_cog = fread(file.path(minidf_path, "psychometric_D2219_ipip_available.csv")) %>%
  mutate(imagid = as.factor(IMAGID))

raw_plasma = fread(file.path(minidf_path, "NfL_pTau_181_CSF_blood_mini_df_newid16.csv")) %>%
  mutate(NFL_CSF = as.numeric(`CSF_NFL (pg/ml)`)) %>%
  mutate(NFL_Plasma = as.numeric(`Plasma_NFL (pg/ml)`)) %>%
  mutate(pTau_Plasma = as.numeric(`Plasma_pTau (pg/ml)`))

raw_plasma$NFL_CSF[raw_plasma$NFL_CSF > 8000] = NA

raw_trem = fread(file.path(minidf_path, "long_ctrem2_and_pgrn_csf_mini_df_newid16.csv")) %>%
  mutate(cTREM2_CSF = as.numeric(`cTREM2_CSF (ng/mL)`)) %>%
  mutate(cTREM2_CSF_adj = as.numeric(`cTREM2_CSF_adj (ng/mL, adjusted to  IS)`)) %>%
  mutate(PGRN_CSF = as.numeric(`PGRN_CSF (ng/mL)`)) %>%
  mutate(PGRN_CSF_adj = as.numeric(`PGRN_CSF_adj (ng/mL, adjusted to IS)`)) %>%
  mutate(newid16.x = as.factor(newid16))

raw_key = fread(file.path(base_path, "data/DIAN/scanner_info/forPete.csv"))%>% 
  mutate(imagid = str_pad(DIANID, 7, pad = '0'))%>%
  dplyr::group_by(imagid) %>% 
  dplyr::slice(1) 
#raw_key$imagid = str_pad(raw_key$DIANID, 7, pad = '0')

raw_qc = fread(file.path(base_path, "data/DIAN/dian_df15_t1_qc/dian_df15_ids_all_qc.csv"))%>%
  separate(Session, c('imagid', 'visit', 'OLD'))%>%
  separate(CNDA, c('source', 'accession', 'freesurfer', 'fs_date'))%>%
  dplyr::select(imagid, visit, fs_date, Scanner, QC)

#merge raw df files
data <- merge(raw_dbn, raw_qc, by = c("imagid", "visit", "fs_date"), all.x = TRUE)%>%
  filter(QC != "FAIL")%>% 
  arrange(dplyr::desc((fs_date))) %>% 
  dplyr::group_by(imagid, visit) %>% 
  dplyr::slice(1) 
data <- merge(data, raw_key, by = c("imagid"), all.x = TRUE)
data <- merge(data, raw_mri, by = c("newid15", "visit"), all.x = TRUE) %>%
  mutate(imagid = as.factor(imagid.y))

data <- merge(data, raw_demo, by = c("imagid", "visit"), all.x = TRUE)
data <- merge(data, raw_clin, by = c("imagid", "visit"), all.x = TRUE)
data <- merge(data, raw_gen, by = c("imagid", "visit"), all.x = TRUE)
data <- merge(data, raw_lumi, by = c("newid16", "visit"), all.x = TRUE)
data <- merge(data, raw_pib, by = c("newid15", "visit"), all.x = TRUE)
data <- merge(data, raw_cog, by = c("newid16", "visit"), all.x = TRUE)
data <- merge(data, raw_plasma, by = c("imagid", "visit"), all.x = TRUE)
data$newid16 = data$newid16.x
data <- merge(data, raw_trem, by = c("newid16", "visit"), all.x = TRUE)
#data <- merge(data, raw_scanner, by = c("newid15", "visit"), all.x = TRUE)

#redefine vars of interest
data$Age = data$VISITAGEc.x
data$Pred_Age_DBN = data$Pred_Age
data$Mutation_Status = recode_factor(data$Mutation.x,  
              '0' = 'Non-Carrier',
              '1' = 'Carrier')

#define EYO
#data$EYO = data$parental_eyo
data$EYO = data$dian_mutpar_eyo

#classify symptomatic/asymptomatic
#data$cdrglob = as.factor(data$cdrglob)
#data_missingmutation = dplyr::filter(data, is.na(Mutation))

data$CDR = as.factor(data$cdrglob.x)

data$Clin_Status = recode_factor(as.factor(data$cdrglob.x),  
              '0' = 'Asymptomatic',
              '0.5' = 'Symptomatic',
              '1' = 'Symptomatic',
              '2' = 'Symptomatic',
              '3' = 'Symptomatic',
              'NA' = 'Missing')

#filter sample
data = subset(data, Age != 'NA') #REMOVE PARTICIPANTS W/O AVAILABLE AGE
data = subset(data, Mutation_Status != 'NA')

#data = subset(data, mr.x != 'mr') #REMOVE REDUNDANT *mr.long* scans

#define APOE categories
data$apoe_fac = as.factor(data$apoe)
data$apoe_pos = recode_factor(data$apoe_fac,
                              '22' = 'E4-',
                              '23' = 'E4-',
                              '24' = 'E4+',
                              '33' = 'E4-',
                              '34' = 'E4+',
                              '44' = 'E4+')
data$apoe_nE4 = recode_factor(data$apoe_fac,
                              '22' = '0',
                              '23' = '0',
                              '24' = '1',
                              '33' = '0',
                              '34' = '1',
                              '44' = '2')

#Education median splot
edu_median = median(data$EDUC.x, na.rm = TRUE)
data$edu_med <- ifelse(data$EDUC.x <= edu_median, "<= Median Education",
                  ifelse(data$EDUC.x > edu_median, "> Median Education", NA
                  ))
table(data$edu_med)

data$edu_12 <- ifelse(data$EDUC.x <= 12, "<= 12 yr Education",
                  ifelse(data$EDUC.x > 12, "> 12 yr Education", NA
                  ))
table(data$edu_12)

#calculate BAG and resid
data$BAG_DBN = data$Pred_Age_DBN - data$Age
data$BAG_DBN_resid_age = lm(data$BAG_DBN ~ data$Age)$residuals

#data$Site = as.factor(data$site.x)
data$Site = recode_factor(data$site.x
                          , '11' = 'A'
                          , '3'  = 'B'
                          , '10' = 'C'
                          , '24' = 'D'
                          , '36' = 'E'
                          , '37' = 'F'
                          , '94' = 'G'
                          , '720' = 'H'
                          , '730' = 'I'
                          , '941' = 'J'
                          , '950' = 'K'
                          , '952' = 'L'
                          , '953' = 'M'
                          , '954' = 'N'
                          , '955' = 'O'
                          , '956' = 'P'
                          , '957' = 'Q'
                          )
#data <- within(data, Site <- relevel(Site, ref = '11'))
data <- within(data, Site <- relevel(Site, ref = 'A'))

intercept = summary(lm(data$Pred_Age_DBN ~ data$Site))$coefficients[1]
data$Pred_Age_DBN_resid_Site = lm(data$Pred_Age_DBN ~ data$Site)$residuals + intercept
data$BAG_DBN_resid_Site = lm(data$BAG_DBN ~ data$Site)$residuals
data$BAG_DBN_resid_age_Site = lm(data$BAG_DBN ~ data$Age * data$Site)$residuals

data$Model = recode_factor(data$Scanner
                           , "DIAN_Achieva" = "Philips_Achieva"
                           , "DIAN_Biograph_mMR" = "Siemens_Biograph"
                           , "DIAN_DISCOVERY_MR750" = "GE_Discovery"
                           , "DIAN_DISCOVERY_MR750w" = "GE_Discovery"
                           , "DIAN_MAGNETOM_Vida" = "Siemens_Vida"
                           , "DIAN_Prisma" = "Siemens_Prisma"
                           , "DIAN_Prisma_fit" = "Siemens_Prisma"
                           , "DIAN_Signa_HDxt" = "GE_Signa"
                           , "DIAN_SIGNA_Premier" = "GE_Signa"
                           , "DIAN_Skyra" = "Siemens_Skyra"
                           , "DIAN_TrioTim" = "Siemens_TrioTim"
                           , "DIAN_TrioTim_NEW_fMRI" = "Siemens_TrioTim"
                           , "DIAN_TrioTim_OLD_fMRI" = "Siemens_TrioTim"
                           , "DIAN_Verio" = "Siemens_Verio"
                           )
table(data$Model)
data$Model = factor(data$Model, levels = c(
  "Siemens_TrioTim", "Siemens_Biograph", "Siemens_Verio", "Siemens_Prisma", "Siemens_Vida", "Siemens_Skyra",
  "Philips_Achieva",
  "GE_Discovery", "GE_Signa"))
data$Manufacturer = recode_factor(data$Model
                           , "Philips_Achieva" = "Philips"
                           , "Siemens_Biograph" = "Siemens"
                           , "GE_Discovery" = "GE"
                           , "GE_Discovery" = "GE"
                           , "Siemens_Vida" = "Siemens"
                           , "Siemens_Prisma" = "Siemens"
                           , "Siemens_Prisma" = "Siemens"
                           , "GE_Signa" = "GE"
                           , "GE_Signa" = "GE"
                           , "Siemens_Skyra" = "Siemens"
                           , "Siemens_TrioTim" = "Siemens"
                           , "Siemens_TrioTim" = "Siemens"
                           , "Siemens_TrioTim" = "Siemens"
                           , "Siemens_Verio" = "Siemens"
                           )
data$Manufacturer = factor(data$Manufacturer, levels = c(
  "Siemens",
  "Philips",
  "GE"))
table(data$Site)
#data$Site = factor(data$Site, levels = c("11", "3", "10", "24", "36", "37", "94",
#                                         "720", "730", "941", "950", "952", "953", 
#                                         "954", "955", "956", "957"))
data$Site = factor(data$Site, levels = c("A", "B", "C", "D", "E", "F", "G",
                                         "H", "I", "J", "K", "L", "M", 
                                         "N", "O", "P", "Q"))
data <- within(data, Model <- relevel(Model, ref = 'Siemens_TrioTim'))

#Regression harmonization
intercept = summary(lm(data$Pred_Age_DBN ~ data$Site + data$Model))$coefficients[1]

data$Pred_Age_DBN_resid_Site_model = lm(data$Pred_Age_DBN ~ data$Site + data$Model)$residuals + intercept

intercept_interaction = summary(lm(data$Pred_Age_DBN ~ data$Site + data$Site:data$Age + data$Model + data$Model:data$Age))$coefficients[1]

data$Pred_Age_DBN_resid_side_model_interaction = lm(data$Pred_Age_DBN ~ data$Site:data$Age + data$Model:data$Age)$residuals + intercept_interaction

data$BAG_DBN_resid_Site_model = lm(data$BAG_DBN ~ data$Site + data$Model)$residuals

data$BAG_DBN_resid_age_Site_model = lm(data$BAG_DBN ~ data$Age * data$Site + data$Model)$residuals

data$BAG_DBN_resid_age_Site_model_interaction = lm(data$BAG_DBN ~ data$Age * data$Site + data$Age * data$Model)$residuals

plot(BAG_DBN_resid_age_Site_model ~ BAG_DBN_resid_age_Site, data = data)
plot(BAG_DBN_resid_age_Site_model_interaction ~ BAG_DBN_resid_age_Site, data = data)

#library(dplyr)
#detach("package:MineICA")
#detach("package:plyr")

data <- data %>% 
  dplyr::group_by(imagid.x)%>%
  arrange(visit)%>%
  mutate(count = "1")%>%
  mutate(visit_mr = cumsum(count))%>%
  mutate(num_visits = max(visit_mr))

#data$visit_mr

#define group
data<- data %>%
  mutate(cdrglobnum = as.numeric(cdrglob.x))%>%
  mutate(group = case_when(
    Mutation.x == 0 & cdrglobnum == 0 ~ "Non-Carrier",
    Mutation.x == 0 & cdrglobnum > 0 ~ "Symptomatic NC",
    Mutation.x == 1 & cdrglobnum == 0 ~ "Asymptomatic MC",
    Mutation.x == 1 & cdrglobnum > 0 ~ "Symptomatic MC"))
data = subset(data, group != "Symptomatic NC")
data$group = fct_relevel(data$group, "Non-Carrier", "Asymptomatic MC", "Symptomatic MC")

#SPLIT MUTATION TYPES (modified from Steph Schultz)

##extract the digits from the FAM_MUTATION_CHG variable
#regular expression to define that you want digits only
require(stringr)

regexp <- "[[:digit:]]+"
#go get the digits from fam_mutation_chg
data$codon_number <- str_extract(data$fam_mutation_chg, regexp)
data$codon_number <- as.numeric(data$codon_number)

#separates PS1 into greater or less than codon 200, leaves APP and PS2 and individual genotypes
data$codon_cat <- as.factor(
  ifelse(data$MUTATIONTYPE == "3", "APP",
                  ifelse(data$MUTATIONTYPE == "2", "PSEN2",
                  ifelse(data$codon_number >= 200, "PSEN1 Codon 200+",
                  ifelse(data$codon_number == 9, "PSEN1 Codon 200+","PSEN1 Codon <200"))))
)

data<- data %>%
  mutate(cdrglobnum = as.numeric(cdrglob.x))%>%
  mutate(codon_cat_sym = case_when(
    codon_cat == "APP" & cdrglobnum == 0 ~ "Asymptomatic APP",
    codon_cat == "APP" & cdrglobnum > 0 ~ "Symptomatic APP",
    codon_cat == "PSEN2" & cdrglobnum == 0 ~ "Asymptomatic PSEN2",
    codon_cat == "PSEN2" & cdrglobnum > 0 ~ "Symptomatic PSEN2", 
    codon_cat == "PSEN1 Codon 200+" & cdrglobnum == 0 ~ "Asymptomatic PSEN1 200+",
    codon_cat == "PSEN1 Codon 200+" & cdrglobnum > 0 ~ "Symptomatic PSEN1 200+",
    codon_cat == "PSEN1 Codon <200" & cdrglobnum == 0 ~ "Asymptomatic PSEN1 <200",
    codon_cat == "PSEN1 Codon <200" & cdrglobnum > 0 ~ "Symptomatic PSEN1 <200"
))

table(data$codon_cat)
#UPDATED CODON CATEGORIZATIONS FROM STEPH SCHULTZ

#codon_stuff.  Extracts the digits from the CHG variable, then classifies the codon number as >200 or <200 for PS1
#does NOT classify APP or PS2 according to codon number

##extract the digits from the FAM_MUTATION_CHG variable
#regular expression to define that you want digits only
require(stringr)

data$j_MutationType <- ifelse(data$MUTATIONTYPE == "1", "PSEN1", ifelse(data$MUTATION == "2", "PSEN2", "APP"))
data$j_MutationType_2 <- paste(data$j_MutationType, data$fam_mutation_chg, sep = "_")


regexp <- "[[:digit:]]+"
#go get the digits from fam_mutation_chg
data$codon_number <- str_extract(data$fam_mutation_chg, regexp)
data$codon_number <- as.numeric(data$codon_number)

#separates PS1 into greater or less than codon 200, leaves APP and PS2 and individual genotypes
data$j_codon_cat <- ifelse(data$MUTATIONTYPE != "1", data$j_MutationType, ifelse(data$codon_number >= 200, "200plus",ifelse(data$codon_number == 9, "200plus","less_than_200")))

table(data$j_codon_cat)


#define imaging measures
##Precuneus thickness
data <- data%>%
  mutate(MR_LR_PRECUNEUS = (MR_LT_PRECUNEUS + MR_RT_PRECUNEUS)/2)
##Hippocampal volume
mean_icv = mean(data$MR_TOTV_INTRACRANIAL.x, na.rm = TRUE)

l_hc_partial = lm(data$MR_LV_HIPPOCAMPUS ~ data$MR_TOTV_INTRACRANIAL.x)
summary(l_hc_partial)
data$mr_lv_hippocampus_corrected = data$MR_LV_HIPPOCAMPUS - (l_hc_partial$coefficients[2] * (data$MR_TOTV_INTRACRANIAL.x - mean_icv))

r_hc_partial = lm(data$MR_RV_HIPPOCAMPUS ~ data$MR_TOTV_INTRACRANIAL.x)
summary(r_hc_partial)
data$mr_rv_hippocampus_corrected = data$MR_RV_HIPPOCAMPUS - (r_hc_partial$coefficients[2] * (data$MR_TOTV_INTRACRANIAL.x - mean_icv))

data$MR_HCV = data$mr_lv_hippocampus_corrected + data$mr_rv_hippocampus_corrected

data$Group = data$group

data_bl = subset(data, visit_mr ==1)
data_long = data
#data_long$visit_mr

data = subset(data, visit_mr ==1)
```

#Remove sites
```{r}
data = subset(data, Site != "B" & Site != "H")
#data$Site = fct_relevel(data$Site, "A"
#, "C", "D", "E", "F", "G"
#, "I", "J", "K", "L", "M"
#, "N", "O", "P", "Q")
data$Site = droplevels(data$Site)


data = subset(data, educ != 'NA')
data = subset(data, codon_cat != 'NA')

#data$apoe_pos = fct_relevel(data$apoe_pos, "E4-", "E4+")
data$apoe_pos = droplevels(data$apoe_pos)
```

#Remove sites long
```{r}
data_long = subset(data_long, Site != "B" & Site != "H")
#data$Site = fct_relevel(data$Site, "A"
#, "C", "D", "E", "F", "G"
#, "I", "J", "K", "L", "M"
#, "N", "O", "P", "Q")
data_long$Site = droplevels(data_long$Site)


data_long = subset(data_long, educ != 'NA')
data_long = subset(data_long, codon_cat != 'NA')

#data_long$apoe_pos = fct_relevel(data_long$apoe_pos, "E4-", "E4+")
data_long$apoe_pos = droplevels(data_long$apoe_pos)
```


#Sites
```{r}
#etaSquared function
my_ES_test <- function(data, variable, by, ...) {
  aovmod = aov(data[[variable]] ~ data[[by]])
  lsr::etaSquared(aovmod)[1,1]
}

# Create summary table, using compact theme
theme_gtsummary_compact()

data$Group = data$group
data$Sex = recode_factor(data$SEX,
                         '2' = 'Female',
                         '1' = 'Male'
                         )
#data$APOE = data$apoe 
#data$APOE = recode_factor(as.factor(data$apoe_pos)
#                          , "E4-" = "E4-"
#                          , "E4+" = "E4+"
#                          , "." = "Unknown"
#)
data$APOE = data$apoe_pos

data$Education = as.numeric(data$EDUC.y)
data$Visits = as.numeric(data$num_visits)
data$CDR = as.factor(data$cdrglob.x)
data$Variant = as.factor(data$codon_cat)

data_bl = subset(data, visit_mr ==1)

data_non = subset(data, Mutation_Status == 'Non-Carrier')

# Define demo table data for Mutation groups
demo_table_data_site <- data_bl%>% 
  select(Site, Age, Sex, Education, Mutation_Status, CDR, EYO, APOE, Variant
         #, Scanner
         #, Visits
         )
demo_table_data_site = subset(demo_table_data_site, select = -c(imagid.x))

# Create demo table
demo_table_site <-demo_table_data_site%>%
  #mutate(Visits = as.numeric(Visits)) %>% 
  tbl_summary(by = Site, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#
    test = list(all_continuous() ~ "kruskal.test" , all_categorical() ~ "fisher.test"), 
    pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  #stat_1 ~ "**A** <br> *({n})*",
                  #Stat_2 ~ "**B** <br> *({n})*",
                  #stat_3 ~ "**C** <br> *({n})*",
                  #stat_4 ~ "**D** <br> *({n})*",
                  #stat_5 ~ "**E** <br> *({n})*",
                  #stat_6 ~ "**F** <br> *({n})*",
                  #stat_7 ~ "**G** <br> *({n})*",
                  #stat_8 ~ "**H** <br> *({n})*",
                  #stat_9 ~ "**I** <br> *({n})*",
                  #stat_10 ~ "**J** <br> *({n})*",
                  #stat_11 ~ "**K** <br> *({n})*",
                  #stat_12 ~ "**L** <br> *({n})*",
                  #stat_13 ~ "**M** <br> *({n})*",
                  #stat_14 ~ "**N** <br> *({n})*",
                  #stat_15 ~ "**O** <br> *({n})*",
                  #stat_16 ~ "**P** <br> *({n})*",
                  #stat_17 ~ "**Q** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**Sites**")%>%
  as_gt()
demo_table_site
gt::gtsave(data = demo_table_site, filename = paste(out_path, "/demo/demographics_site_", ver_date, ".html", sep = "")) 

table(data$Site)
table(data_non$Site)

hist(table(data$Site), breaks = 50)

```
#Models
```{r}
#etaSquared function
my_ES_test <- function(data, variable, by, ...) {
  aovmod = aov(data[[variable]] ~ data[[by]])
  lsr::etaSquared(aovmod)[1,1]
}

# Create summary table, using compact theme
theme_gtsummary_compact()

data$Group = data$group
data$Sex = recode_factor(data$SEX,
                         '2' = 'Female',
                         '1' = 'Male'
                         )
#data$APOE = data$apoe 
#data$APOE = recode_factor(as.factor(data$apoe_pos)
#                          , "E4-" = "E4-"
#                          , "E4+" = "E4+"
#                          , "." = "Unknown"
#)
data$APOE = data$apoe_pos

data$Education = as.numeric(data$EDUC.y)
data$Visits = as.numeric(data$num_visits)
data$CDR = as.factor(data$cdrglob.x)
data$Variant = as.factor(data$codon_cat)

data_bl = subset(data, visit_mr ==1)

# Define demo table data for Mutation groups
demo_table_data_Model <- data_bl%>% 
  select(Model, Age, Sex, Education, Mutation_Status, CDR, EYO, APOE, Variant
         #, Visits
         )
demo_table_data_Model = subset(demo_table_data_Model, select = -c(imagid.x))

# Create demo table
demo_table_Model <-demo_table_data_Model%>%
  #mutate(Visits = as.numeric(Visits)) %>% 
  tbl_summary(by = Model, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#test = list(all_continuous() ~ "aov" , all_categorical() ~ "kruskal.test"), 
        pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  #stat_1 ~ "**A** <br> *({n})*",
                  #Stat_2 ~ "**B** <br> *({n})*",
                  #stat_3 ~ "**C** <br> *({n})*",
                  #stat_4 ~ "**D** <br> *({n})*",
                  #stat_5 ~ "**E** <br> *({n})*",
                  #stat_6 ~ "**F** <br> *({n})*",
                  #stat_7 ~ "**G** <br> *({n})*",
                  #stat_8 ~ "**H** <br> *({n})*",
                  #stat_9 ~ "**I** <br> *({n})*",
                  #stat_10 ~ "**J** <br> *({n})*",
                  #stat_11 ~ "**K** <br> *({n})*",
                  #stat_12 ~ "**L** <br> *({n})*",
                  #stat_13 ~ "**M** <br> *({n})*",
                  #stat_14 ~ "**N** <br> *({n})*",
                  #stat_15 ~ "**O** <br> *({n})*",
                  #stat_16 ~ "**P** <br> *({n})*",
                  #stat_17 ~ "**Q** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**Models**")%>%
  as_gt()
demo_table_Model
gt::gtsave(data = demo_table_Model, filename = paste(out_path, "/demo/demographics_Model_", ver_date, ".html", sep = "")) 

table(data$Model)
table(data_non$Model)

hist(table(data$Model), breaks = 50)

```


#Combat xsection
```{r}
#Combat harmoniztion
##Set up factors and models
features = data.frame(cbind(data$Pred_Age_DBN
                            , data$BAG_DBN
                            , data$MR_HCV
                            , data$MR_LR_PRECUNEUS
                            , data$CortSig_Thickness
                            ))
dat = t(as.matrix(features))

batch = as.factor(data$Site)
levels(batch)
table(batch)

mod = model.matrix(~ data$Age 
                   + as.factor(data$SEX)
                   + data$educ
                   + data$group
                   #+ data$Mutation_Status
                   #+ as.factor(data$CDR)
                   + data$EYO
                   + data$apoe_pos
                   + data$codon_cat
                   )
#head(mod)

##Harmonize with ComBAT
combat.harmonized = neuroCombat(dat=dat, batch=batch, mod=mod
                                 #, mean.only = FALSE
                                 )

names(combat.harmonized)

dat.combat = combat.harmonized$dat.combat

data$Pred_Age_DBN_ComBat = t(dat.combat)[,1]

data$MR_HCV = t(dat.combat)[,3]
data$MR_LR_PRECUNEUS = t(dat.combat)[,4]
data$CortSig_Thickness = t(dat.combat)[,5]

#data$BAG_DBN_ComBat = t(dat.combat)[,2]
data$BAG_DBN_ComBat = data$Pred_Age_DBN_ComBat - data$Age
data$BAG_DBN_ComBat_resid_age = lm(data$BAG_DBN_ComBat ~ data$Age)$residuals
#MODIFIED 10/5/23 TO ACCOUNT FOR DIFFERENT SLOPES PER GROUP - REMOVES GROUP DIF
#data$BAG_DBN_ComBat_resid_age = lm(data$BAG_DBN_ComBat ~ data$Group:data$Age)$residuals



##Check Model params
###Gamma

drawPriorGamma(combat.harmonized$estimates
               #, xlim = c(-1, 1)
               #, col = c(#"green","blue")
               )

###Delta Hat

drawPriorDelta(combat.harmonized$estimates
               #, xlim = c(-2, 2)
               #, col = c("green", "black")
               )

#subset sample
data_non = subset(data, Mutation_Status == 'Non-Carrier')
data_car = subset(data, Mutation_Status == 'Carrier')

#define linear corrected Pred_Age and BAG using BAG slope (behesti et al, 2019)
lm_DBN_non = lm(BAG_DBN_ComBat ~ Age, data = data_non)
summary(lm_DBN_non)
lm_DBN_non_B = lm_DBN_non$coefficients[1]
lm_DBN_non_A = lm_DBN_non$coefficients[2]
data <- data%>%
  mutate(Pred_Age_DBN_lin = Pred_Age_DBN_ComBat - ((lm_DBN_non_A*Age) + lm_DBN_non_B))
plot(data$Pred_Age_DBN_lin ~ data$BAG_DBN_ComBat)
data$BAG_DBN_lin = data$Pred_Age_DBN_lin - data$Age

#define linear corrected Pred_Age and BAG using (Cole et al 2018 (Mortality), De Lange & Cole 2020)
lm_delange_non = lm(Pred_Age_DBN_ComBat ~ Age, data = data_non)
summary(lm_delange_non)
lm_delange_non_int = lm_delange_non$coefficients[1]
lm_delange_non_slope = lm_delange_non$coefficients[2]
data <- data%>%
  mutate(Pred_Age_DBN_delange = (Pred_Age_DBN_ComBat - lm_delange_non_int) / lm_delange_non_slope)
plot(data$Pred_Age_DBN_delange ~ data$BAG_DBN_ComBat)
data$BAG_DBN_delange = data$Pred_Age_DBN_delange - data$Age

plot(data$Pred_Age_DBN_delange ~ data$Pred_Age_DBN_lin)
plot(data$BAG_DBN_delange ~ data$BAG_DBN_lin)

plot(data$Pred_Age_DBN_delange ~ data$Age)

cor(data$Pred_Age_DBN_delange, data$Age)
cor(data$Pred_Age_DBN_ComBat, data$Age)


#define linear corrected Pred_Age and BAG using (Cole et al 2018 (Mortality), De Lange & Cole 2020)
#for unharmonized data
lm_delange_unharm_non = lm(Pred_Age_DBN ~ Age, data = data_non)
summary(lm_delange_unharm_non)
lm_delange_unharm_non_int = lm_delange_unharm_non$coefficients[1]
lm_delange_unharm_non_slope = lm_delange_unharm_non$coefficients[2]
data <- data%>%
  mutate(Pred_Age_DBN_delange_unharm = (Pred_Age_DBN - lm_delange_unharm_non_int) / lm_delange_unharm_non_slope)
plot(data$Pred_Age_DBN_delange_unharm ~ data$BAG_DBN)
data$BAG_DBN_delange_unharm = data$Pred_Age_DBN_delange_unharm - data$Age



```



#Combat long
```{r}
#Combat harmoniztion
##Set up factors and models
features_long = data.frame(cbind(data_long$Pred_Age_DBN
                            , data_long$BAG_DBN                            
                            , data_long$MR_HCV
                            , data_long$MR_LR_PRECUNEUS
                            , data_long$CortSig_Thickness
                            ))

dat_long = t(as.matrix(features_long))

batch_long = as.factor(data_long$Site)
levels(batch_long)
table(batch_long)

mod_long = model.matrix(~ data_long$Age 
                   + as.factor(data_long$SEX)
                   + data_long$educ
                   + data_long$group
                   #+ data_long$Mutation_Status
                   #+ as.factor(data_long$CDR)
                   + data_long$EYO
                   + data_long$apoe_pos
                   + data_long$codon_cat
                   )
#head(mod)

##Harmonize with ComBAT
combat.harmonized_long = neuroCombat(dat=dat_long, batch=batch_long, mod=mod_long
                                 #, mean.only = FALSE
                                 )

names(combat.harmonized_long)

dat.combat_long = combat.harmonized_long$dat.combat

data_long$Pred_Age_DBN_ComBat = t(dat.combat_long)[,1]

data_long$MR_HCV = t(dat.combat_long)[,3]
data_long$MR_LR_PRECUNEUS = t(dat.combat_long)[,4]
data_long$CortSig_Thickness = t(dat.combat_long)[,5]

#data_long$BAG_DBN_ComBat = t(dat.combat_long)[,2]
data_long$BAG_DBN_ComBat = data_long$Pred_Age_DBN_ComBat - data_long$Age
data_long$BAG_DBN_ComBat_resid_age = lm(data_long$BAG_DBN_ComBat ~ data_long$Age)$residuals


##Check Model params
###Gamma

drawPriorGamma(combat.harmonized_long$estimates
               #, xlim = c(-1, 1)
               #, col = c(#"green","blue")
               )

###Delta Hat

drawPriorDelta(combat.harmonized_long$estimates
               #, xlim = c(-2, 2)
               #, col = c("green", "black")
               )

data_long$Pred_Age_Choice = data_long$Pred_Age_DBN_ComBat
data_long$BAG_Choice = data_long$BAG_DBN_ComBat_resid_age

```

##Regression dilution
###Uncorrected
```{r}

cor_age_uncorrected = ggplot(data, aes(y = BAG_DBN_ComBat, x = Age)) +
  geom_point(aes(color = Group)
    ) +
  labs(title="Reression Dilution Effect (Uncorrected)", y = "BAG (Uncorrected)", x = "Age") +
  geom_smooth(aes(color = Group, fill = Group), 
              method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(Group = imagid.x, color = Group)
            ) + 
  #scale_fill_discrete(labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample')) +
  scale_color_manual(values = c("blue", "orange", "red", "black")
                     , labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "orange", "red", "grey"), labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = Group), label.y = c(-23, -26,-29)) +
  theme_bw(base_size = 15) +
  theme(legend.position="bottom", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_age_uncorrected

ggsave(paste(out_path, '/cor_age/dian_df15_cor_BAG_Uncorrected_Age_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```
###Resid Age
```{r}

cor_age_resid = ggplot(data, aes(y = BAG_DBN_ComBat_resid_age, x = Age)) +
  geom_point(aes(color = Group)
    ) +
  labs(title="Reression Dilution Effect (Residual)", y = "BAG (Age-Corrected)", x = "Age") +
  geom_smooth(aes(color = Group, fill = Group), 
              method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(Group = imagid.x, color = Group)
            ) + 
  #scale_fill_discrete(labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample')) +
  scale_color_manual(values = c("blue", "orange", "red", "black")
                     , labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "orange", "red", "grey"), labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = Group), label.y = c(-23, -26,-29)) +
  theme_bw(base_size = 15) +
  theme(legend.position="bottom", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_age_resid

ggsave(paste(out_path, '/cor_age/dian_df15_cor_BAG_residage_Age_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```
###Linear
```{r}

cor_age_linear = ggplot(data, aes(y = BAG_DBN_lin, x = Age)) +
  geom_point(aes(color = Group)
    ) +
  labs(title="Reression Dilution Effect (Linear)", y = "BAG (Linear)", x = "Age") +
  geom_smooth(aes(color = Group, fill = Group), 
              method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(Group = imagid.x, color = Group)
            ) + 
  #scale_fill_discrete(labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample')) +
  scale_color_manual(values = c("blue", "orange", "red", "black")
                     , labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "orange", "red", "grey"), labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = Group), label.y = c(-23, -26,-29)) +
  theme_bw(base_size = 15) +
  theme(legend.position="bottom", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_age_linear

ggsave(paste(out_path, '/cor_age/dian_df15_cor_BAG_linear_Age_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```

###delange
```{r}

cor_age_delange = ggplot(data, aes(y = BAG_DBN_delange, x = Age)) +
  geom_point(aes(color = Group)
    ) +
  labs(title="Reression Dilution Effect (Corrected)", y = "BAG (Corrected)", x = "Age") +
  geom_smooth(aes(color = Group, fill = Group), 
              method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(Group = imagid.x, color = Group)
            ) + 
  #scale_fill_discrete(labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample')) +
  scale_color_manual(values = c("blue", "orange", "red", "black")
                     , labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "orange", "red", "grey"), labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = Group), label.y = c(-23, -26,-29)) +
  theme_bw(base_size = 15) +
  theme(legend.position="bottom", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_age_delange

ggsave(paste(out_path, '/cor_age/dian_df15_cor_BAG_delange_Age_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```

####Merge Age fig
```{r}
ggarrange(
  cor_age_uncorrected
  #, cor_age_resid, cor_age_linear
  , cor_age_delange
  , labels = c("A", "B"
               #, "C", "D"
               )
  , ncol = 2
  #, nrow = 2
  , common.legend = TRUE, legend = "bottom"
)

ggsave(paste(ms_path, '/FigureS1_RegressionDilution_', ver_date, '.png', sep = ""), width = 10, height = 5, units = "in", dpi = 320)
```

#Post combat
```{r}
#check scanner info
data$Model = (as.factor(data$Model))

#define baseline and longitudinal sessions
data <- data %>% 
  group_by(imagid.x)%>%
  arrange(visit)%>%
  mutate(count = "1")%>%
  mutate(visit_mr = cumsum(count))%>%
  mutate(num_visits = max(visit_mr))
#plot(num_visits ~ visit_mr, data = data)

#define csf ratios
data$CSF_AB4240_ratio = as.numeric(data$LUMIPULSE_CSF_AB42.x) / data$LUMIPULSE_CSF_AB40.x
data$CSF_ptauab42_ratio = as.numeric(data$LUMIPULSE_CSF_pTau.x) / as.numeric(data$LUMIPULSE_CSF_AB42.x)
data$CSF_ptauab40_ratio = as.numeric(data$LUMIPULSE_CSF_pTau.x) / data$LUMIPULSE_CSF_AB40.x
data$CSF_ttauab40_ratio = as.numeric(data$LUMIPULSE_CSF_tTau.x) / data$LUMIPULSE_CSF_AB40.x

#subset sample
data_non = subset(data, Mutation_Status == 'Non-Carrier')
data_car = subset(data, Mutation_Status == 'Carrier')

data_bl = subset(data, visit_mr ==1)
data_non_bl = subset(data_bl, Mutation_Status == 'Non-Carrier')
data_car_bl = subset(data_bl, Mutation_Status == 'Carrier')

#define cog compoSite
##ANIMALS
m_ANIMALS <- mean(data_non_bl$ANIMALS, na.rm = T)
sd_ANIMALS <- sd(data_non_bl$ANIMALS, na.rm = T)
data$z_ANIMALS <- (data$ANIMALS - m_ANIMALS)/sd_ANIMALS
##WAIS
m_WAIS <- mean(data_non_bl$WAIS, na.rm = T)
sd_WAIS <- sd(data_non_bl$WAIS, na.rm = T)
data$z_WAIS <- (data$WAIS - m_WAIS)/sd_WAIS
##TRAILB
m_TRAILB <- mean(data_non_bl$TRAILB, na.rm = T)
sd_TRAILB <- sd(data_non_bl$TRAILB, na.rm = T)
data$z_TRAILB <- (data$TRAILB - m_TRAILB)/sd_TRAILB
##MEMUNITS
m_MEMUNITS <- mean(data_non_bl$MEMUNITS.x, na.rm = T)
sd_MEMUNITS <- sd(data_non_bl$MEMUNITS.x, na.rm = T)
data$z_MEMUNITS <- (data$MEMUNITS.x - m_MEMUNITS)/sd_MEMUNITS

data <- data%>%
  mutate(global_cog = (z_ANIMALS
                       + z_WAIS
                       + (-1*z_TRAILB)
                       + z_MEMUNITS) / 4)

data_cog = data[,which(names(data)=="z_ANIMALS"):which(names(data)=="global_cog")]
hist(data_cog$global_cog, breaks = 50)

pairs.panels(data_cog)

#define group
data<- data %>%
  mutate(cdrglobnum = as.numeric(cdrglob.x))%>%
  mutate(group = case_when(
    Mutation.x == 0 & cdrglobnum == 0 ~ "Non-Carrier",
    Mutation.x == 0 & cdrglobnum > 0 ~ "Symptomatic NC",
    Mutation.x == 1 & cdrglobnum == 0 ~ "Asymptomatic MC",
    Mutation.x == 1 & cdrglobnum > 0 ~ "Symptomatic MC"))
data = subset(data, group != "Symptomatic NC")
data$group = fct_relevel(data$group, "Non-Carrier", "Asymptomatic MC", "Symptomatic MC")
#data$group = factor(data$group, levels = c("Non-Carrier", "Asymptomatic MC", "Symptomatic MC"))

#SPLIT MUTATION TYPES (modified from Steph Schultz)

##extract the digits from the FAM_MUTATION_CHG variable
#regular expression to define that you want digits only
require(stringr)

regexp <- "[[:digit:]]+"
#go get the digits from fam_mutation_chg
data$codon_number <- str_extract(data$fam_mutation_chg, regexp)
data$codon_number <- as.numeric(data$codon_number)

#separates PS1 into greater or less than codon 200, leaves APP and PS2 and individual genotypes
data$codon_cat <- as.factor(
  ifelse(data$MUTATIONTYPE == "3", "APP",
                  ifelse(data$MUTATIONTYPE == "2", "PSEN2",
                  ifelse(data$codon_number >= 200, "PSEN1 Codon 200+",
                  ifelse(data$codon_number == 9, "PSEN1 Codon 200+","PSEN1 Codon <200"))))
)

data<- data %>%
  mutate(cdrglobnum = as.numeric(cdrglob.x))%>%
  mutate(codon_cat_sym = case_when(
    codon_cat == "APP" & cdrglobnum == 0 ~ "Asymptomatic APP",
    codon_cat == "APP" & cdrglobnum > 0 ~ "Symptomatic APP",
    codon_cat == "PSEN2" & cdrglobnum == 0 ~ "Asymptomatic PSEN2",
    codon_cat == "PSEN2" & cdrglobnum > 0 ~ "Symptomatic PSEN2", 
    codon_cat == "PSEN1 Codon 200+" & cdrglobnum == 0 ~ "Asymptomatic PSEN1 200+",
    codon_cat == "PSEN1 Codon 200+" & cdrglobnum > 0 ~ "Symptomatic PSEN1 200+",
    codon_cat == "PSEN1 Codon <200" & cdrglobnum == 0 ~ "Asymptomatic PSEN1 <200",
    codon_cat == "PSEN1 Codon <200" & cdrglobnum > 0 ~ "Symptomatic PSEN1 <200"
))

table(data$codon_cat)
#UPDATED CODON CATEGORIZATIONS FROM STEPH SCHULTZ

#codon_stuff.  Extracts the digits from the CHG variable, then classifies the codon number as >200 or <200 for PS1
#does NOT classify APP or PS2 according to codon number

##extract the digits from the FAM_MUTATION_CHG variable
#regular expression to define that you want digits only
require(stringr)

data$j_MutationType <- ifelse(data$MUTATIONTYPE == "1", "PSEN1", ifelse(data$MUTATION == "2", "PSEN2", "APP"))
data$j_MutationType_2 <- paste(data$j_MutationType, data$fam_mutation_chg, sep = "_")


regexp <- "[[:digit:]]+"
#go get the digits from fam_mutation_chg
data$codon_number <- str_extract(data$fam_mutation_chg, regexp)
data$codon_number <- as.numeric(data$codon_number)

#separates PS1 into greater or less than codon 200, leaves APP and PS2 and individual genotypes
data$j_codon_cat <- ifelse(data$MUTATIONTYPE != "1", data$j_MutationType, ifelse(data$codon_number >= 200, "200plus",ifelse(data$codon_number == 9, "200plus","less_than_200")))

table(data$j_codon_cat)

#second gene_cat: classifying PSEN1 mutations according to topology - cytoplasmic/lumenal/intramembrane/transmembrane domains
#utilizes information from uniprot 
#For PSEN1: www.uniprot.org/uniprot/P49768)
#For APP: https://www.uniprot.org/uniprot/P05067
#For PSEN2: https://www.uniprot.org/uniprot/P49810
#Step 1: categorizes PSEN1 carriers, if not PSEN1 carriers ,then carries over gene_cat_2
#updated on Aug 3 2018
PSEN_topo_type <- function(x) 	{
  PSEN1_topo_type <- 	ifelse(data$MUTATIONTYPE == "1",
                             ifelse(x <=82 & x!= 9 & x!=4, "PSEN1_cytoplasmic",(
                               ifelse(x == 9, "PSEN1_cytoplasmic",(
                                 ifelse(x == 4, "PSEN1_undefined",(
                                   ifelse(x >= 83 & x <= 103, "PSEN1_transmembrane",(
                                     ifelse(x >= 104 & x <= 132, "PSEN1_undefined",(
                                       ifelse(x >= 133 & x <= 153, "PSEN1_transmembrane",(
                                         ifelse(x >= 154 & x <= 166, "PSEN1_cytoplasmic",(
                                           ifelse(x >= 167 & x <= 189, "PSEN1_transmembrane",(
                                             ifelse(x >= 190 & x <= 194, "PSEN1_undefined",(
                                               ifelse(x >= 195 & x <= 216, "PSEN1_transmembrane",(
                                                 ifelse(x >= 217 & x <= 220, "PSEN1_cytoplasmic",(
                                                   ifelse(x >= 221 & x <= 241, "PSEN1_transmembrane",(
                                                     ifelse(x >= 242 & x <= 248, "PSEN1_undefined",(
                                                       ifelse(x >= 249 & x <= 272, "PSEN1_transmembrane",(
                                                         ifelse(x >= 273 & x <= 380, "PSEN1_cytoplasmic",(
                                                           ifelse(x >= 381 & x <= 401, "PSEN1_transmembrane",(
                                                             ifelse(x >= 402 & x <= 407, "PSEN1_lumenal",(
                                                               ifelse(x >= 408 & x <= 428, "PSEN1_transmembrane",(
                                                                 ifelse(x >=429 & x <= 432, "PSEN1_cytoplasmic",(
                                                                   ifelse(x >= 433 & x <= 453, "PSEN1_transmembrane",(
                                                                     ifelse(x >= 454 & x <= 467, "PSEN1_undefined","not_PSEN1"))))))))))))))))))))))))))))))))))))))))),
                             "not_PSEN1")
}

#NOTES ON ABOVE:
#del_exon 9 should not be included in cyto 1, hence x!= 9 above in cyto 1 categorization 
#Intron 4: IVS4+7A should not be included in cyto 1, hence x!=4 above in cyto 1 categorization 

data$PSEN_topo_type <- as.factor(PSEN_topo_type(data$codon_number))

table(data$PSEN_topo_type)

#recode for NC, PSEN1 CY, PSEN1 TM, or other (not_PSEN1)
data$PSEN_topo_type_final_nc <- ifelse(data$Mutation.x !=1, "NC", (
  ifelse(data$PSEN_topo_type == "PSEN1_cytoplasmic","PSEN1_cytoplasmic", (
    ifelse(data$PSEN_topo_type == "PSEN1_transmembrane","PSEN1_transmembrane", (
      ifelse(data$PSEN_topo_type == "not_PSEN1","not_PSEN1","PSEN1_undefined")))))))

table(data$PSEN_topo_type_final_nc)

#filter out "not_PSEN1" variants
data_PSEN1_CY_TM <- subset(data, PSEN_topo_type_final_nc != "not_PSEN1") 

table(data_PSEN1_CY_TM$PSEN_topo_type_final_nc)



data = subset(data, group != "Symptomatic NC")

data$Sex = recode_factor(data$SEX,
                         '1' = 'Male',
                         '2' = 'Female')

data$EYO2 = data$EYO^2


############################
# CHOOSE PRED & BAG MEASURES
############################
#data$Pred_Age_Choice = data$Pred_Age_DBN_resid_side_model_interaction
#data$BAG_Choice = data$BAG_DBN_resid_age_Site_model_interaction
#data$Pred_Age_Choice = data$Pred_Age_DBN_ComBat
#data$BAG_Choice = data$BAG_DBN_ComBat_resid_age

data$Pred_Age_Choice = data$Pred_Age_DBN_ComBat
data$BAG_Choice = as.numeric(data$BAG_DBN_delange)

hist(data$BAG_DBN_ComBat_resid_age)

##################################
# REMOVE LONGITUDINAL DATA
##################################
data_bl = subset(data, visit_mr ==1)
#data_long = data
data = data_bl

data_share = data%>%
  dplyr::select(visit, newid16, BAG_DBN_ComBat_resid_age)

#subset sample AGAIN
data_non = subset(data, Mutation_Status == 'Non-Carrier')
data_car = subset(data, Mutation_Status == 'Carrier')


data_bl = subset(data, visit_mr ==1)
data_non_bl = subset(data_bl, Mutation_Status == 'Non-Carrier')
data_car_bl = subset(data_bl, Mutation_Status == 'Carrier')

table(data_car$codon_cat)

data_car_codon = subset(data_car, codon_cat != 'NA')
data_car_codon = subset(data_car_codon, codon_cat != 'PSEN2')

write.csv(data_share, paste(out_path, '/dian_bagvals_', ver_date, '.csv', sep = ""), row.names = FALSE)

```

#Inspect data
##BAG
All DIAN
```{r}
BAG_DBN_sd = sd(data$BAG_DBN)
BAG_DBN_m = mean(data$BAG_DBN)
hist(data$BAG_DBN, breaks = 50,
     xlab = "BAG_DBN", main = "All DIAN")
abline(v = BAG_DBN_m + (3*BAG_DBN_sd), col = 'red')
abline(v = BAG_DBN_m - (3*BAG_DBN_sd), col = 'red')

BAG_DBN_resid_sd = sd(data$BAG_Choice)
BAG_DBN_resid_m = mean(data$BAG_Choice)
hist(data$BAG_Choice, breaks = 50,
     xlab = "BAG_DBN_Resid", main = "All DIAN")
abline(v = BAG_DBN_resid_m + (3*BAG_DBN_resid_sd), col = 'red')
abline(v = BAG_DBN_resid_m - (3*BAG_DBN_resid_sd), col = 'red')

```




Non-carriers
```{r}
BAG_DBN_sd = sd(data_non$BAG_DBN)
BAG_DBN_m = mean(data_non$BAG_DBN)
hist(data_non$BAG_DBN, breaks = 50,
     xlab = "BAG_DBN", main = "Non-Carriers")
abline(v = BAG_DBN_m + (3*BAG_DBN_sd), col = 'red')
abline(v = BAG_DBN_m - (3*BAG_DBN_sd), col = 'red')

BAG_DBN_resid_sd = sd(data_non$BAG_Choice)
BAG_DBN_resid_m = mean(data_non$BAG_Choice)
hist(data_non$BAG_Choice, breaks = 50,
     xlab = "BAG_DBN_Resid", main = "Non-Carriers")
abline(v = BAG_DBN_resid_m + (3*BAG_DBN_resid_sd), col = 'red')
abline(v = BAG_DBN_resid_m - (3*BAG_DBN_resid_sd), col = 'red')

```

###Site effects - uncorrected
Check Normality
```{r}
# Compute Shapiro wilk test 
shapiro.test(data_non$BAG_DBN_delange_unharm)
#W = 0.98858, p-value = 0.1588

# Draw a qq plot by group
qq_site_uncorrected = ggqqplot(data_non, x = "BAG_DBN_delange_unharm", facet.by = "Site", main = "Site Differences (Unharmonized)")
qq_site_uncorrected
```
Check equality of variance
```{r}
leveneTest(BAG_DBN_delange_unharm ~ Site, data = data_non)
#Levene's Test for Homogeneity of Variance (center = median)
#       Df F value Pr(>F)
#group  14  0.8447 0.6199
#      164  
```

```{r}
kruskal.test(BAG_DBN_delange_unharm ~ Site, data = data_non)
summary(aov(BAG_DBN_delange_unharm ~ Site, data = data_non))

table(data$Site)
table(data_non$Site)

boxplot_bag_site_uncorrected = ggplot(data_non, aes(x=Site, y=BAG_DBN_delange_unharm, fill=Site)) +
  labs(title="Site Differences (Unharmonized)",x="Site", y = "BAG (Age-Corrected)") +
  geom_violin(trim=FALSE, color = "black") + 
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  #stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
  #                   , aes(label = ..p.adj..)
  #                   )+ # Add pairwise comparisons p-value
  stat_compare_means(label.y = 25, label.x = 1
                     #, method = "anova"
                     )+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5))

boxplot_bag_site_uncorrected

```
###Model effects - uncorrected
Check Normality
```{r}
# Compute Shapiro wilk test 
shapiro.test(data_non$BAG_DBN_delange_unharm)
#W = 0.98858, p-value = 0.1588

# Draw a qq plot by group
qq_scanner_uncorrected = ggqqplot(data_non, x = "BAG_DBN_delange_unharm", facet.by = "Model", main = "Model Differences (Unharmonized)")
qq_scanner_uncorrected
```
Check equality of variance
```{r}
leveneTest(BAG_DBN_delange_unharm ~ Model, data = data_non)
#Levene's Test for Homogeneity of Variance (center = median)
#       Df F value Pr(>F)
#group   8  0.9869  0.448
#      170  
```

```{r}
kruskal.test(BAG_DBN_delange_unharm ~ Model, data = data_non)
summary(aov(BAG_DBN_delange_unharm ~ Model, data = data_non))
table(data$Model)
table(data_non$Model)

boxplot_bag_Model_uncorrected = ggplot(data_non, aes(x=Model, y=BAG_DBN_delange_unharm, fill=Model)) +
  labs(title="Scanner Differences (Unharmonized)",x="Model", y = "BAG (Age-Corrected)") +
  geom_violin(trim=FALSE, color = "black") + 
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  #stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
  #                   , aes(label = ..p.adj..)
  #                   )+ # Add pairwise comparisons p-value
  stat_compare_means(label.y = 25, label.x = 1
                     #, method = "anova"
                     )+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5)
        , axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)
        )

boxplot_bag_Model_uncorrected

```

###Site effects - corrected
Check Normality
```{r}
# Compute Shapiro wilk test
shapiro.test(data_non$BAG_Choice)
#W = 0.99073, p-value = 0.3011

# Draw a qq plot by group
qq_site_corrected = ggqqplot(data_non, x = "BAG_Choice", facet.by = "Site", main = "Site Differences (Harmonized)")
qq_site_corrected
```
Check equality of variance
```{r}
leveneTest(BAG_Choice ~ Site, data = data_non)
#Levene's Test for Homogeneity of Variance (center = median)
#       Df F value Pr(>F)
#group  14     0.7 0.7722
#      164 
```

```{r}
kruskal.test(BAG_Choice ~ Site, data = data_non)
summary(aov(BAG_Choice ~ Site, data = data_non))

boxplot_bag_site_corrected = ggplot(data_non, aes(x=Site, y=BAG_Choice, fill=Site)) +
  labs(title="Site Differences (Harmonized)",x="Site", y = "BAG (Age-Corrected)") +
  geom_violin(trim=FALSE, color = "black") + 
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  #stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
  #                   , aes(label = ..p.adj..)
  #                   )+ # Add pairwise comparisons p-value
  stat_compare_means(label.y = 25, label.x = 1
                     #, method = "anova"
                     )+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5))

boxplot_bag_site_corrected

```
###Model effects - corrected
Check Normality
```{r}
# Compute Shapiro wilk test
shapiro.test(data_non$BAG_Choice)
#W = 0.99073, p-value = 0.3011

# Draw a qq plot by group
qq_model_corrected = ggqqplot(data_non, x = "BAG_Choice", facet.by = "Model", main = "Scanner Differences (Harmonized)")
qq_model_corrected
```

Check equality of variance
```{r}
leveneTest(BAG_Choice ~ Model, data = data_non)
#Levene's Test for Homogeneity of Variance (center = median)
#       Df F value Pr(>F)
#group   8  1.0291 0.4161
#      170  
```

```{r}
kruskal.test(BAG_Choice ~ Model, data = data_non)
summary(aov(BAG_Choice ~ Model, data = data_non))

boxplot_bag_Model_corrected = ggplot(data_non, aes(x=Model, y=BAG_Choice, fill=Model)) +
  labs(title="Scanner Differences (Harmonized)",x="Model", y = "BAG (Age-Corrected)") +
  geom_violin(trim=FALSE, color = "black") + 
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  #stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
  #                   , aes(label = ..p.adj..)
  #                   )+ # Add pairwise comparisons p-value
  stat_compare_means(label.y = 25, label.x = 1
                     #, method = "anova"
                     )+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5)
        , axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)
        )

boxplot_bag_Model_corrected

```

Carriers
```{r}
BAG_DBN_sd = sd(data_car$BAG_DBN)
BAG_DBN_m = mean(data_car$BAG_DBN)
hist(data_car$BAG_DBN, breaks = 50,
     xlab = "BAG_DBN", main = "Carriers")
abline(v = BAG_DBN_m + (3*BAG_DBN_sd), col = 'red')
abline(v = BAG_DBN_m - (3*BAG_DBN_sd), col = 'red')

BAG_DBN_resid_sd = sd(data_car$BAG_Choice)
BAG_DBN_resid_m = mean(data_car$BAG_Choice)
hist(data_car$BAG_Choice, breaks = 50,
     xlab = "BAG_DBN_Resid", main = "Carriers")
abline(v = BAG_DBN_resid_m + (3*BAG_DBN_resid_sd), col = 'red')
abline(v = BAG_DBN_resid_m - (3*BAG_DBN_resid_sd), col = 'red')

```
##Demographics long

```{r}
data_long$Group = data_long$group
data_long$Sex = recode_factor(data_long$SEX,
                         '2' = 'Female',
                         '1' = 'Male'
                         )
#data_long$APOE = data_long$apoe 
#data_long$APOE = recode_factor(as.factor(data_long$apoe_pos)
#                          , "E4-" = "E4-"
#                          , "E4+" = "E4+"
#                          , "." = "Unknown"
#)
data_long$APOE = data_long$apoe_pos

data_long$Education = as.numeric(data_long$EDUC.y)
data_long$Visits = as.numeric(data_long$num_visits)
data_long$CDR = as.factor(data_long$cdrglob.x)
data_long$Variant = as.factor(data_long$codon_cat)

data_long$Race = recode_factor(as.factor(data_long$RACE.y)
                          , "Asian" = "Asian"
                          , "Black or African American" = "Black or African American"
                          , "Native Hawaiian or Other Pacific Islander" = "Native Hawaiian or Other Pacific Islander"
                          , "White" = "White"
                          , "Other (specify)" = "Other"
                          , "Unknown" = "Unknown"
                          , "NA"  = "Unknown"
                          #, "" = "Unknown"
                          )

data_long = data_long%>%
  mutate(Race = case_when(
    Race == "Other" ~ RACEX.y
    , Race != "Other" ~ Race
  ))%>%
  mutate(Race = case_when(
    RACESEC.y == "American Indian or Alaska Native" | RACESEC.y == "Asian" | RACESEC.y == "Black or African American" | RACESEC.y == "Other (specify)" | RACESEC.y == "White" ~ "More than one race"
    , RACESEC.y != "American Indian or Alaska Native" & RACESEC.y != "Asian" & RACESEC.y != "Black or African American" & RACESEC.y != "Other (specify)" & RACESEC.y != "White" ~ Race
    #RACESEC.y != "None Reported" & RACESEC.y != "Unkown" ~ "More than one race"
    #, RACESEC.y == "None Reported" | RACESEC.y == "Unkown" ~ Race
  ))
table(data_long$Race)

data_long$Race = recode_factor(as.factor(data_long$Race)
                          , "Aboriginal" = "Aboriginal"
                          , "Aboriginal Australian" = "Aboriginal"
                          , "Australian aboriginal" = "Aboriginal"
                          , "Arab" = "Arab"
                          , "Asian" = "Asian"
                          , "Black or African American" = "Black or African American"
                          , "Croatian" = "Unknown"
                          , "Hispanic" = "Hispanic or Latinx"
                          , "Latin" = "Hispanic or Latinx"
                          , "LATIN" = "Hispanic or Latinx"
                          , "Mexican American" = "Hispanic or Latinx"
                          , "Latin American" = "Hispanic or Latinx"
                          , "Libyan, Middle Eastern" = "Middle Eastern or North African"
                          , "Mediterranean, Middle Eastern" = "Middle Eastern or North African"
                          , "North African" = "Middle Eastern or North African"
                          , "Native Hawaiian or Other Pacific Islander" = "Native Hawaiian or Other Pacific Islander"
                          , "White" = "White"
                          , "More than one race" = "More than one race"
                          , "Other (specify)" = "Other"
                          , "Unknown" = "Unknown"
                          , "none" = "Unknown"
                          , "NA"  = "Unknown"
                          #, "" = "Unknown"
                          )



data_retest_long_date = data_long%>% 
  select(num_visits, visit_mr
         , mr_date
         )

hist(data_retest_long_date$num_visits)

#data_retest_long = subset(data_retest_long, num_visits > 2)

data_retest_wide_date = spread(data_retest_long_date, key = visit_mr
                          #, value = Pred_Age_DBN
                          #, value = Pred_Age_DBN_ComBat
                          , value = mr_date
                          )


data_retest_wide_date = data_retest_wide_date[,c(1,3:4)]
data_retest_wide_date = data_retest_wide_date[complete.cases(data_retest_wide_date), ]

colnames(data_retest_wide_date) = c("imagid.x", "t1", "t2")


  
data_retest_wide_date$Time = time_length(difftime(
  as.Date(data_retest_wide_date$t2, format='%m/%d/%Y')
         , as.Date(data_retest_wide_date$t1, format='%m/%d/%Y')
  ), "years")

describe(data_retest_wide_date$Time)

data_long_demo = merge(data_retest_wide_date, data_long, by = "imagid.x", all.x = TRUE)
#data_long_demo = subset(data_long_demo, visit == 'v00')

data_long_demo = data_long_demo[!duplicated(subset(data_long_demo, select = c( "imagid.x"
                                                                               #, "visit"
                                                                               ))), ]

# Define demo table data for Mutation groups
demo_table_data1_long <- data_long_demo%>% 
  select(Mutation_Status, Time, Age, Sex, Education
         #, Race
         , CDR, EYO, APOE, Variant
         #, Visits
         )
#demo_table_data1_long = subset(demo_table_data1_long, select = -c(imagid.x))

# Create demo table
demo_table1_long <-demo_table_data1_long%>%
  #mutate(Visits = as.numeric(Visits)) %>% 
  tbl_summary(by = Mutation_Status, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#test = list(all_continuous() ~ "aov" , all_categorical() ~ "kruskal.test"), 
        pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  stat_1 ~ "**Non-Carriers** <br> *({n})*",
                  stat_2 ~ "**Carriers** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**ADAD Mutation Groups**")%>%
  as_gt()
demo_table1_long
gt::gtsave(data = demo_table1_long, filename = paste(out_path, "/demo/demographics_long_mutation_", ver_date, ".html", sep = "")) 


# Define demo table data for Mutation & symptomatic groups
demo_table_data2_long <- data_long_demo%>% 
  select(Group, Time, Age, Sex, Education
         #, Race
         , CDR, EYO, APOE, Variant
         #, Visits
         )
#demo_table_data1_long = subset(demo_table_data1_long, select = -c(imagid.x))

# Create demo table
demo_table2_long <-demo_table_data2_long%>%
  #mutate(Visits = as.numeric(Visits)) %>% 
  tbl_summary(by = Group, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#test = list(all_continuous() ~ "aov" , all_categorical() ~ "kruskal.test"), 
        pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  stat_1 ~ "**Non-Carriers** <br> *({n})*",
                  stat_2 ~ "**Asymptomatic Carriers** <br> *({n})*",
                  stat_3 ~ "**Symptomatic Carriers** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**ADAD Mutation Groups**")%>%
  as_gt()
demo_table2_long
gt::gtsave(data = demo_table2_long, filename = paste(out_path, "/demo/demographics_long_mutationsym_", ver_date, ".html", sep = "")) 


```

##Reliability
###BAG
```{r}
data_retest_long_bag = data_long%>% 
  select(num_visits, visit_mr
         #, Pred_Age_DBN
         #, Pred_Age_DBN_ComBat
         , BAG_Choice
         #, Mutation_Status
         )

hist(data_retest_long_bag$num_visits)

#data_retest_long = subset(data_retest_long, num_visits > 2)

data_retest_wide_bag = spread(data_retest_long_bag, key = visit_mr
                          #, value = Pred_Age_DBN
                          #, value = Pred_Age_DBN_ComBat
                          , value = BAG_Choice
                          )


data_retest_wide_bag = data_retest_wide_bag[,3:4]
data_retest_wide_bag = data_retest_wide_bag[complete.cases(data_retest_wide_bag), ]


icc_bag = irr::icc(data_retest_wide_bag
              , model = "twoway"
              , type = "agreement"
              , unit = "single"
              )
icc_bag

colnames(data_retest_wide_bag) = c("t1", "t2")

scale = data_retest_wide_bag[,1:2]

scatter_retest_bag = ggplot(data_retest_wide_bag, aes(x = t1, y = t2)) +
  geom_point(aes()) +
  labs(title="Reliability: BAG",x="Time 1", y = "Time 2") +
  geom_smooth(aes(), method = lm) + 
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(scale), max(scale)) + 
  ylim(min(scale), max(scale)) + 
  annotate("text"
           , x = (quantile(seq(min(scale, na.rm = TRUE), max(scale, na.rm = TRUE)), .9))
           , y = (min(scale, na.rm = TRUE))
           , label = paste("italic(ICC) == ", round(icc_bag$value, 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        )

scatter_retest_bag
```
###Brain Age
```{r}
data_retest_long_brainage = data_long%>% 
  select(num_visits, visit_mr
         #, Pred_Age_DBN
         , Pred_Age_DBN_ComBat
         #, BAG_Choice
         )

hist(data_retest_long_brainage$num_visits)

#data_retest_long = subset(data_retest_long, num_visits > 2)

data_retest_wide_brainage = spread(data_retest_long_brainage, key = visit_mr
                          #, value = Pred_Age_DBN
                          , value = Pred_Age_DBN_ComBat
                          #, value = BAG_Choice
                          )

data_retest_wide_brainage = data_retest_wide_brainage[,3:4]
data_retest_wide_brainage = data_retest_wide_brainage[complete.cases(data_retest_wide_brainage), ]

icc_brainage = irr::icc(data_retest_wide_brainage
              , model = "twoway"
              , type = "agreement"
              , unit = "single"
              )
icc_brainage

colnames(data_retest_wide_brainage) = c("t1", "t2")

scale = data_retest_wide_brainage[,1:2]

scatter_retest_brainage = ggplot(data_retest_wide_brainage, aes(x = t1, y = t2)) +
  geom_point(aes()) +
  labs(title="Reliability: Brain Age",x="Time 1", y = "Time 2") +
  geom_smooth(aes(), method = lm) + 
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(scale), max(scale)) + 
  ylim(min(scale), max(scale)) + 
  annotate("text"
           , x = (quantile(seq(min(scale, na.rm = TRUE), max(scale, na.rm = TRUE)), .9))
           , y = (min(scale, na.rm = TRUE))
           , label = paste("italic(ICC) == ", round(icc_brainage$value, 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        )

scatter_retest_brainage
```
###Hippocampus Volume
```{r}
data_retest_long_hcv = data_long%>% 
  select(num_visits, visit_mr
         #, Pred_Age_DBN
         , MR_HCV
         #, BAG_Choice
         )

hist(data_retest_long_hcv$num_visits)

#data_retest_long = subset(data_retest_long, num_visits > 2)

data_retest_wide_hcv = spread(data_retest_long_hcv, key = visit_mr
                          #, value = Pred_Age_DBN
                          , value = MR_HCV
                          #, value = BAG_Choice
                          )

data_retest_wide_hcv = data_retest_wide_hcv[,3:4]
data_retest_wide_hcv = data_retest_wide_hcv[complete.cases(data_retest_wide_hcv), ]

icc_hcv = irr::icc(data_retest_wide_hcv
              , model = "twoway"
              , type = "agreement"
              , unit = "single"
    )
icc_hcv

colnames(data_retest_wide_hcv) = c("t1", "t2")

scale = data_retest_wide_hcv[,1:2]

scatter_retest_hcv = ggplot(data_retest_wide_hcv, aes(x = t1, y = t2)) +
  geom_point(aes()) +
  labs(title="Reliability: Hippocampus Volume",x="Time 1", y = "Time 2") +
  geom_smooth(aes(), method = lm) + 
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(scale), max(scale)) + 
  ylim(min(scale), max(scale)) + 
  annotate("text"
           , x = (quantile(seq(min(scale, na.rm = TRUE), max(scale, na.rm = TRUE)), .9))
           , y = (min(scale, na.rm = TRUE))
           , label = paste("italic(ICC) == ", round(icc_hcv$value, 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        )

scatter_retest_hcv
```

###Precuneus Thickness
```{r}
data_retest_long_precuneus = data_long%>% 
  select(num_visits, visit_mr
         #, Pred_Age_DBN
         , MR_LR_PRECUNEUS
         #, BAG_Choice
         )

hist(data_retest_long_precuneus$num_visits)

#data_retest_long = subset(data_retest_long, num_visits > 2)

data_retest_wide_precuneus = spread(data_retest_long_precuneus, key = visit_mr
                          #, value = Pred_Age_DBN
                          , value = MR_LR_PRECUNEUS
                          #, value = BAG_Choice
                          )

data_retest_wide_precuneus = data_retest_wide_precuneus[,3:4]
data_retest_wide_precuneus = data_retest_wide_precuneus[complete.cases(data_retest_wide_precuneus), ]

icc_precuneus = irr::icc(data_retest_wide_precuneus
              , model = "twoway"
              , type = "agreement"
              , unit = "single"
    )
icc_precuneus

colnames(data_retest_wide_precuneus) = c("t1", "t2")

scale = data_retest_wide_precuneus[,1:2]

scatter_retest_precuneus = ggplot(data_retest_wide_precuneus, aes(x = t1, y = t2)) +
  geom_point(aes()) +
  labs(title="Reliability: Precuneus Thickness",x="Time 1", y = "Time 2") +
  geom_smooth(aes(), method = lm) + 
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(scale), max(scale)) + 
  ylim(min(scale), max(scale)) + 
  annotate("text"
           , x = (quantile(seq(min(scale, na.rm = TRUE), max(scale, na.rm = TRUE)), .9))
           , y = (min(scale, na.rm = TRUE))
           , label = paste("italic(ICC) == ", round(icc_precuneus$value, 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        )

scatter_retest_precuneus
```

###Cortical Signature Thickness
```{r}
data_retest_long_cortsig = data_long%>% 
  select(num_visits, visit_mr
         #, Pred_Age_DBN
         , CortSig_Thickness
         #, BAG_Choice
         )

hist(data_retest_long_cortsig$num_visits)

#data_retest_long = subset(data_retest_long, num_visits > 2)

data_retest_wide_cortsig = spread(data_retest_long_cortsig, key = visit_mr
                          #, value = Pred_Age_DBN
                          , value = CortSig_Thickness
                          #, value = BAG_Choice
                          )

data_retest_wide_cortsig = data_retest_wide_cortsig[,3:4]
data_retest_wide_cortsig = data_retest_wide_cortsig[complete.cases(data_retest_wide_cortsig), ]

icc_cortsig = irr::icc(data_retest_wide_cortsig
              , model = "twoway"
              , type = "agreement"
              , unit = "single"
    )
icc_cortsig

colnames(data_retest_wide_cortsig) = c("t1", "t2")

scale = data_retest_wide_cortsig[,1:2]

scatter_retest_cortsig = ggplot(data_retest_wide_cortsig, aes(x = t1, y = t2)) +
  geom_point(aes()) +
  labs(title="Reliability: Cortical Signature Thickness",x="Time 1", y = "Time 2") +
  geom_smooth(aes(), method = lm) + 
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(scale), max(scale)) + 
  ylim(min(scale), max(scale)) + 
  annotate("text"
           , x = (quantile(seq(min(scale, na.rm = TRUE), max(scale, na.rm = TRUE)), .9))
           , y = (min(scale, na.rm = TRUE))
           , label = paste("italic(ICC) == ", round(icc_cortsig$value, 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        )

scatter_retest_cortsig
```

##Demographic tables

```{r demo-output, echo = FALSE, warning = FALSE, message = FALSE}
#etaSquared function
my_ES_test <- function(data, variable, by, ...) {
  aovmod = aov(data[[variable]] ~ data[[by]])
  lsr::etaSquared(aovmod)[1,1]
}

# Create summary table, using compact theme
theme_gtsummary_compact()

data$Group = data$group
data$Sex = recode_factor(data$SEX,
                         '2' = 'Female',
                         '1' = 'Male'
                         )
#data$APOE = data$apoe 
#data$APOE = recode_factor(as.factor(data$apoe_pos)
#                          , "E4-" = "E4-"
#                          , "E4+" = "E4+"
#                          , "." = "Unknown"
#)
table(data$RACE.y)
table(data$RACESEC.y)
data$Race = recode_factor(as.factor(data$RACE.y)
                          , "Asian" = "Asian"
                          , "Black or African American" = "Black or African American"
                          , "Native Hawaiian or Other Pacific Islander" = "Native Hawaiian or Other Pacific Islander"
                          , "White" = "White"
                          , "Other (specify)" = "Other"
                          , "Unknown" = "Unknown"
                          , "NA"  = "Unknown"
                          #, "" = "Unknown"
                          )

data = data%>%
  mutate(Race = case_when(
    Race == "Other" ~ RACEX.y
    , Race != "Other" ~ Race
  ))%>%
  mutate(Race = case_when(
    RACESEC.y == "American Indian or Alaska Native" | RACESEC.y == "Asian" | RACESEC.y == "Black or African American" | RACESEC.y == "Other (specify)" | RACESEC.y == "White" ~ "More than one race"
    , RACESEC.y != "American Indian or Alaska Native" & RACESEC.y != "Asian" & RACESEC.y != "Black or African American" & RACESEC.y != "Other (specify)" & RACESEC.y != "White" ~ Race
    #RACESEC.y != "None Reported" & RACESEC.y != "Unkown" ~ "More than one race"
    #, RACESEC.y == "None Reported" | RACESEC.y == "Unkown" ~ Race
  ))
table(data$Race)

data$Race = recode_factor(as.factor(data$Race)
                          , "Aboriginal" = "Aboriginal"
                          , "Aboriginal Australian" = "Aboriginal"
                          , "Australian aboriginal" = "Aboriginal"
                          , "Arab" = "Arab"
                          , "Asian" = "Asian"
                          , "Black or African American" = "Black or African American"
                          , "Croatian" = "Unknown"
                          , "Hispanic" = "Hispanic or Latinx"
                          , "Latin" = "Hispanic or Latinx"
                          , "LATIN" = "Hispanic or Latinx"
                          , "Mexican American" = "Hispanic or Latinx"
                          , "Latin American" = "Hispanic or Latinx"
                          , "Libyan, Middle Eastern" = "Middle Eastern or North African"
                          , "Mediterranean, Middle Eastern" = "Middle Eastern or North African"
                          , "North African" = "Middle Eastern or North African"
                          , "Native Hawaiian or Other Pacific Islander" = "Native Hawaiian or Other Pacific Islander"
                          , "White" = "White"
                          , "More than one race" = "More than one race"
                          , "Other (specify)" = "Other"
                          , "Unknown" = "Unknown"
                          , "none" = "Unknown"
                          , "NA"  = "Unknown"
                          #, "" = "Unknown"
                          )

table(data$Race)
data_lme_apoe = subset(data_lme, apoe != '.')
data_lme_apoe = subset(data_lme_apoe, codon_cat != 'NA')
data_lme_apoe$apoe_cat = as.factor(data_lme_apoe$apoe_pos)
data_lme_apoe = data_lme_apoe%>%
  mutate(mut_apoe = case_when(
    Mutation_Status == "Non-Carrier" & apoe_cat == "E4-" ~ "Non-Carrier / E4-",
    Mutation_Status == "Non-Carrier" & apoe_cat == "E4+" ~ "Non-Carrier / E4+",
    Mutation_Status == "Carrier" & apoe_cat == "E4-" ~ "Carrier / E4-",
    Mutation_Status == "Carrier" & apoe_cat == "E4+" ~ "Carrier / E4+"))

data_other = subset(data, Race == "Other" | Race == "Unknown" | Race == "")
data_other_race = data_other[, c("Race", "RACE.x", "RACESEC.x", "RACESECX.x", "RACETER.x", "RACETERX.x", "RACEX.x", "RACE.y", "RACEX.y", "RACESEC.y", "RACESECX.y", "RACETER.y", "RACETERX.y")]



data$Education = as.numeric(data$EDUC.y)
data$Visits = as.numeric(data$num_visits)
data$PIB_SUVR = data$PIB_fSUVR_rsf_TOT_CORTMEAN
data$CDR = as.factor(data$cdrglob.x)
data$Variant = as.factor(data$codon_cat)

data_bl = subset(data, visit_mr ==1)

# Define demo table data for Mutation groups
demo_table_data1 <- data_bl%>% 
  select(Mutation_Status, Age, Sex, Education, Race, CDR, EYO, APOE, Variant
         #, Visits
         )
demo_table_data1 = subset(demo_table_data1, select = -c(imagid.x))

# Create demo table
demo_table1 <-demo_table_data1%>%
  #mutate(Visits = as.numeric(Visits)) %>% 
  tbl_summary(by = Mutation_Status, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#test = list(all_continuous() ~ "aov" , all_categorical() ~ "kruskal.test"), 
        pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  stat_1 ~ "**Non-Carriers** <br> *({n})*",
                  stat_2 ~ "**Carriers** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**ADAD Mutation Groups**")%>%
  as_gt()
demo_table1
gt::gtsave(data = demo_table1, filename = paste(out_path, "/demo/demographics_mutation_", ver_date, ".html", sep = "")) 

# Define demo table data for Mutation & symptomatic groups
demo_table_data2 <- data_bl%>% 
  select(Group, Age, Sex, Education, Race, CDR, EYO, APOE, Variant
         #, Visits
         )
demo_table_data2 = subset(demo_table_data2, select = -c(imagid.x))

# Create demo table
demo_table2 <-demo_table_data2%>%
  tbl_summary(by = Group, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#test = list(all_continuous() ~ "aov" , all_categorical() ~ "kruskal.test"), 
    pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  stat_1 ~ "**Non-Carriers** <br> *({n})*",
                  stat_2 ~ "**Asymptomatic Carriers** <br> *({n})*",
                  stat_3 ~ "**Symptomatic Carriers** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**ADAD Mutation Groups**")%>%
  as_gt()
demo_table2
gt::gtsave(data = demo_table2, filename = paste(out_path, "/demo/demographics_mutationsym_", ver_date, ".html", sep = "")) 

data_car_bl = subset(data_bl, Mutation_Status == "Carrier")
data_car_bl$Group = droplevels(data_car_bl$Group)

# Define demo table data for Mutation & symptomatic groups
demo_table_data3 <- data_car_bl%>% 
  select(Group, Age, Sex, Education, Race, CDR, EYO, APOE
         #, Visits
         )
demo_table_data3 = subset(demo_table_data3, select = -c(imagid.x))

# Create demo table
demo_table3 <-demo_table_data3%>%
  #mutate(Visits = as.numeric(Visits)) %>% 
  tbl_summary(by = Group, #missing = "always", missing_text = "Missing",
                          type = all_continuous() ~ "continuous2",
                          statistic = all_continuous() ~ c("{mean} ({sd})"),
                          digits = c(Age ~ c(1,1), Education ~ c(1,1)
                                     , EYO ~ c(1,1)))%>%
  add_stat(fns = all_continuous() ~ my_ES_test) %>%
  add_p(#test = list(all_continuous() ~ "aov" , all_categorical() ~ "kruskal.test"), 
    pvalue_fun = ~style_pvalue(.x,digits =2))%>% 
  add_n() %>%
  modify_header(update = list(
                  label ~ "",
                  n ~ "*N*",
                  stat_1 ~ "**Asymptomatic Carriers** <br> *({n})*",
                  stat_2 ~ "**Symptomatic Carriers** <br> *({n})*",
                  add_stat_1 ~ "**\U03B7\U00B2**"
                  ))%>% 
  bold_labels()%>%
  italicize_levels()%>%
  modify_spanning_header(all_stat_cols()~"**ADAD Symptom Groups**")%>%
  as_gt()
demo_table3
gt::gtsave(data = demo_table3, filename = paste(out_path, "/demo/demographics_sym_", ver_date, ".html", sep = "")) 

```

#Test model fit
##Uncorrected
###Site effects - uncorrected
```{r}
yscale = data$Pred_Age_DBN

scatter_fit_non_Site = ggplot(data_non, aes(x = Age, y = Pred_Age_DBN)) +
  geom_point(aes(color = Site)) +
  labs(title="Site Differences (Unharmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(col = 'darkgray', method = lm, linetype = 'longdash') + 
  geom_smooth(aes(color = Site, fill = Site), method = lm, se = FALSE) + 
  geom_line(aes(group = imagid.x, color = Site)) + 
  #scale_color_manual(values = c("blue")) +
  #scale_fill_manual(values = c("blue")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  theme_bw(base_size = 15) +
  theme(legend.position=c(.9,.4), 
    plot.title = element_text(hjust = 0.5)#, legend.title = element_blank()
    , legend.key = element_blank(), legend.background=element_blank()
            , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank()) +
  guides(fill = FALSE)
scatter_fit_non_Site

ggsave(paste(out_path, '/model_fit/dian_df15_fit_non_Site_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```

###model effects - uncorrected
```{r}
yscale = data$Pred_Age_DBN

scatter_fit_non_model = ggplot(data_non, aes(x = Age, y = Pred_Age_DBN)) +
  geom_point(aes(color = Model)) +
  labs(title="Scanner Differences (Unharmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(col = 'darkgray', method = lm, linetype = 'longdash') + 
  geom_smooth(aes(color = Model, fill = Model), method = lm, se = FALSE) + 
  geom_line(aes(group = imagid.x, color = Model)) + 
  #scale_color_manual(values = c("blue")) +
  #scale_fill_manual(values = c("blue")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  theme_bw(base_size = 15) +
  theme(legend.position=c(.2, .75), plot.title = element_text(hjust = 0.5)#, legend.title = element_blank()
        , legend.key = element_blank(), legend.background=element_blank()
                , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank())

scatter_fit_non_model

ggsave(paste(out_path, '/model_fit/dian_df15_fit_non_model_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```
##Harmonized
###Non Carriers
```{r}
yscale = c(data_non$Pred_Age_Choice, data_non$Age)

scatter_fit_non_harmonized = ggplot(data_non, aes(x = Age, y = Pred_Age_Choice)) +
  geom_point(color = "blue") +
  labs(title="DBN-Predicted Brain Age (Harmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(color = "blue", fill = "blue", method = lm) + 
  geom_line(aes(group = imagid.x), color = "blue") + 
  #scale_color_manual(values = c("blue")) +
  #scale_fill_manual(values = c("blue")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale))-4, max(c(data$Age, yscale))+4) + 
  ylim(min(c(data$Age, yscale))-4, max(c(data$Age, yscale))+4) + 
  theme_bw(base_size = 15) +
    annotate("text", x = max(yscale)-((max(yscale)-min(yscale))*0.1)+4, y = min(yscale)+((max(yscale)-min(yscale))*0.12)-4, color = "blue", label = paste("italic(r) == ", round(cor(data_non$Age, data_non$Pred_Age_Choice), 2)),
  parse = TRUE) +
  annotate("text", x = max(yscale)-((max(yscale)-min(yscale))*0.1)+4, y = min(yscale)+((max(yscale)-min(yscale))*0.08)-4, color = "blue", label = paste("italic(R) ^ 2 == ", round(cor(data_non$Age, data_non$Pred_Age_Choice)^2, 2)),
  parse = TRUE) +
  annotate("text", x = max(yscale)-((max(yscale)-min(yscale))*0.1)+4, y = min(yscale)+((max(yscale)-min(yscale))*0.04)-4, color = "blue", label = paste("italic(MAE) == ", round(MAE(data_non$Age, data_non$Pred_Age_Choice), 2)),
  parse = TRUE) +
  annotate("text", x = max(yscale)-((max(yscale)-min(yscale))*0.1)+4, y = min(yscale)+((max(yscale)-min(yscale))*0.00)-4, color = "blue", label = paste("italic(RMSE) == ", round(RMSE(data_non$Age, data_non$Pred_Age_Choice), 2)),
  parse = TRUE) +
  theme(legend.position=c(.2, .85), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()        
        , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank())

scatter_fit_non_harmonized

ggsave(paste(out_path, '/model_fit/dian_df15_fit_non_harmonized_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

test_r2_harm = cor(data_non$Pred_Age_Choice, data_non$Age)^2
test_mae_harm = mae(data_non$Pred_Age_Choice, data_non$Age)
test_rmse_harm = rmse(data_non$Pred_Age_Choice, data_non$Age)

```


###Site effects - corrected
```{r}
yscale = data$Pred_Age_Choice

scatter_fit_non_Site_correctSite = ggplot(data_non, aes(x = Age, y = Pred_Age_Choice)) +
  geom_point(aes(color = Site)) +
  labs(title="Site Differences (Harmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(col = 'darkgray', method = lm, linetype = 'longdash') + 
  geom_smooth(aes(color = Site, fill = Site), method = lm, se = FALSE) + 
  geom_line(aes(group = imagid.x, color = Site)) + 
  #scale_color_manual(values = c("blue")) +
  #scale_fill_manual(values = c("blue")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  theme_bw(base_size = 15) +
  theme(legend.position=c(.9,.4), plot.title = element_text(hjust = 0.5)#, legend.title = element_blank()
        , legend.key = element_blank(), legend.background=element_blank()
                , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank()) +
  guides(fill = FALSE)

scatter_fit_non_Site_correctSite

ggsave(paste(out_path, '/model_fit/dian_df15_fit_non_Site_Sitecorrected_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)


```


###Model effects - corrected
```{r}
yscale = data$Pred_Age_Choice

scatter_fit_non_model_correctSite = ggplot(data_non, aes(x = Age, y = Pred_Age_Choice)) +
  geom_point(aes(color = Model)) +
  labs(title="Scanner Differences (Harmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(col = 'darkgray', method = lm, linetype = 'longdash') + 
  geom_smooth(aes(color = Model, fill = Model), method = lm, se = FALSE) + 
  geom_line(aes(group = imagid.x, color = Model)) + 
  #scale_color_manual(values = c("blue")) +
  #scale_fill_manual(values = c("blue")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  theme_bw(base_size = 15) +
  theme(legend.position=c(.2, .75), plot.title = element_text(hjust = 0.5)#, legend.title = element_blank()
        , legend.key = element_blank(), legend.background=element_blank()
                , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank())

scatter_fit_non_model_correctSite

ggsave(paste(out_path, '/model_fit/dian_df15_fit_non_model_Sitecorrected_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)


```

##Merge Site Comparison
```{r}
#install.packages()
ggarrange(
  ggarrange(scatter_fit_non_Site, boxplot_bag_site_uncorrected
            , scatter_fit_non_Site_correctSite, boxplot_bag_site_corrected
            , nrow = 2, ncol = 2
            , widths = c(0.5, 1)
            , labels = c("A", "B", "C", "D"))
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(ms_path, '/FigureS1_SiteComparison_', ver_date, '.png', sep = ""), width = 18, height = 12, units = "in", dpi = 320)

table(data$Site)
table(data_non$Site)
```
##Merge Model Comparison
```{r}
#install.packages()
ggarrange(
  ggarrange(scatter_fit_non_model, boxplot_bag_Model_uncorrected
            , scatter_fit_non_model_correctSite, boxplot_bag_Model_corrected
            , nrow = 2, ncol = 2
            , widths = c(0.5, 1)
            , labels = c("A", "B", "C", "D"))
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(ms_path, '/FigureS2_ModelComparison_', ver_date, '.png', sep = ""), width = 18, height = 12, units = "in", dpi = 320)

table(data$Model)
table(data_non$Model)
```


#Test group differences
##Check assumptions
Check Normality
```{r}
# Compute Shapiro wilk test
shapiro.test(data$BAG_Choice)
#W = 0.95203, p-value = 1.081e-10

# Draw a qq plot by group
qq_site_corrected = ggqqplot(data, x = "BAG_Choice"
                             , facet.by = "Group"
                             , main = "Distribution of BAG Values")
qq_site_corrected
ggsave(paste(ms_path, '/FigureS4_QQPlot_', ver_date, '.png', sep = ""), width = 12, height = 4, units = "in", dpi = 320)
```
Check equality of variance
```{r}
leveneTest(BAG_Choice ~ Group, data = data)
#Levene's Test for Homogeneity of Variance (center = median)
#       Df F value Pr(>F)
#group   2   10.67 2.997e-05 ***
#      433 
```

##non-carrier vs. asymptomatic vs symptomatic
###Scatterplot + NC performance
```{r}
yscale = data$Pred_Age_Choice

annotation <- data.frame(
   x = c(76, 76, 76, 76),
   y = c(26, 23, 20, 17),
   label = c(
     paste("r = ", round(cor(data_non$Age, data_non$Pred_Age_Choice), 2))
           , paste("R^2 = ", round(cor(data_non$Age, data_non$Pred_Age_Choice)^2, 2))
           , paste("MAE = ", round(MAE(data_non$Age, data_non$Pred_Age_Choice), 2))
           , paste("RMSE = ", round(RMSE(data_non$Age, data_non$Pred_Age_Choice), 2))
)
)

scatter_dif_mutationsym_performance = ggplot(data, aes(x = Age, y = Pred_Age_Choice)) +
  geom_point(aes(color = group)) +
  labs(title="DBN-Predicted Brain Age (Harmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("blue", "orange", "red")) +
  scale_fill_manual(values = c("blue", "orange", "red")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #geom_text(data=annotation, aes( x=x, y=y, label=label)                 
  #         , color="blue"
  #         #, size=7
  #         #, fontface="italic" 
  #         ) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.16), color = "blue", fontface="bold", label = ("Accuracy in NCs")) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.12), color = "blue", label = paste("italic(r) == ", round(cor(data_non$Age, data_non$Pred_Age_Choice), 2)), parse = TRUE) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.08), color = "blue", label = paste("italic(R) ^ 2 == ", round(cor(data_non$Age, data_non$Pred_Age_Choice)^2, 2)), parse = TRUE) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.04), color = "blue", label = paste("italic(MAE) == ", round(MAE(data_non$Age, data_non$Pred_Age_Choice), 2)), parse = TRUE) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.00), color = "blue", label = paste("italic(RMSE) == ", round(RMSE(data_non$Age, data_non$Pred_Age_Choice), 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank()
        )

scatter_dif_mutationsym_performance

ggsave(paste(out_path, '/dif_mutation/dian_df15_dif_mutationsym_scatter_performance_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

test_r2_non = cor(data_non$Pred_Age_Choice, data_non$Age)^2
test_mae_non = mae(data_non$Pred_Age_Choice, data_non$Age)
test_rmse_non = rmse(data_non$Pred_Age_Choice, data_non$Age)

test_r2_car = cor(data_car$Pred_Age_Choice, data_car$Age)^2
test_mae_car = mae(data_car$Pred_Age_Choice, data_car$Age)
test_rmse_car = rmse(data_car$Pred_Age_Choice, data_car$Age)
```

###Scatterplot only
```{r}
yscale = data$Pred_Age_Choice

scatter_dif_mutationsym = ggplot(data, aes(x = Age, y = Pred_Age_Choice)) +
  geom_point(aes(color = group)) +
  labs(title="DBN-Predicted Brain Age (Harmonized)",x="True Age", y = "Predicted Age") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("blue", "orange", "red")) +
  scale_fill_manual(values = c("blue", "orange", "red")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale))-4, max(c(data$Age, yscale))+1) + 
  ylim(min(c(data$Age, yscale))-4, max(c(data$Age, yscale))+1) + 
  theme_bw(base_size = 15) +
  theme(legend.position=c(.8, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank()
        )

scatter_dif_mutationsym

ggsave(paste(out_path, '/dif_mutation/dian_df15_dif_mutationsym_scatter_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

test_r2_non = cor(data_non$Pred_Age_Choice, data_non$Age)^2
test_mae_non = mae(data_non$Pred_Age_Choice, data_non$Age)
test_rmse_non = rmse(data_non$Pred_Age_Choice, data_non$Age)

test_r2_car = cor(data_car$Pred_Age_Choice, data_car$Age)^2
test_mae_car = mae(data_car$Pred_Age_Choice, data_car$Age)
test_rmse_car = rmse(data_car$Pred_Age_Choice, data_car$Age)
```
###Violin plot
```{r}
kruskal.test(BAG_Choice ~ Group, data = data)

data_lme = subset(data, educ != 'NA')

compare_means(BAG_Choice ~ group,  data = data, method = "wilcox.test"
              , p.adjust.method = "fdr")
my_comparisons <- list( c("Non-Carrier", "Asymptomatic MC")
                        , c("Asymptomatic MC", "Symptomatic MC")
                        , c("Non-Carrier", "Symptomatic MC") )

violin_dif_mutationsym = ggplot(data, aes(x=group, y=BAG_Choice, fill=group)) +
  labs(title="DBN-BAG (Harmonized)",x="", y = "BAG (Age-Corrected)") +
  geom_violin(trim=FALSE, color = "black") + 
  scale_fill_manual(breaks=c("Non-Carrier", "Asymptomatic MC", "Symptomatic MC"),
                              values=c("blue", "orange", "red")) +

  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  #stat_compare_means()+ # Add omnibus test
  stat_compare_means(comparisons = my_comparisons
                     , method = "wilcox.test"
                     , aes(label = ..p.adj..)
                     )+ # Add pairwise comparisons p-value
  #stat_compare_means(label.y = 50)+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5))

violin_dif_mutationsym

ggsave(paste(out_path, '/dif_mutation/dian_df15_dif_mutationsym_violin_', ver_date, '.png', sep = ""), width = 5, height = 5, units = "in", dpi = 320)
```
###Linear regression
```{r}
full_lm_dbn = lm(data = data,
            BAG_Choice ~ 
              group
            + Sex
            + educ
            + apoe_pos
            )
summary(full_lm_dbn)
as.matrix(etasq(full_lm_dbn))
```
###Effect sizes
```{r}
data_sym = subset (data_car, group == "Symptomatic MC")
data_nonsym = subset (data, group != "Asymptomatic MC")

cohen.d(as.numeric(data_nonsym$BAG_Choice), data_nonsym$group)
```

##Age-Corrected performance
Scatterplot + performance
```{r}
yscale = data$Pred_Age_DBN_delange

annotation <- data.frame(
   x = c(76, 76, 76, 76),
   y = c(26, 23, 20, 17),
   label = c(
     paste("r = ", round(cor(data_non$Age, data_non$Pred_Age_DBN_delange), 2))
           , paste("R^2 = ", round(cor(data_non$Age, data_non$Pred_Age_DBN_delange)^2, 2))
           , paste("MAE = ", round(MAE(data_non$Age, data_non$Pred_Age_DBN_delange), 2))
           , paste("RMSE = ", round(RMSE(data_non$Age, data_non$Pred_Age_DBN_delange), 2))
)
)

scatter_dif_mutationsym_performance_agecorrected = ggplot(data, aes(x = Age, y = Pred_Age_DBN_delange)) +
  geom_point(aes(color = group)) +
  labs(title="DBN-Predicted Brain Age (Harmonized & Age-Corrected)",x="True Age", y = "Predicted Age") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("blue", "orange", "red")) +
  scale_fill_manual(values = c("blue", "orange", "red")) +
  geom_abline(slope = 1, intercept = 0, linetype = "longdash") + 
  xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #geom_text(data=annotation, aes( x=x, y=y, label=label)                 
  #         , color="blue"
  #         #, size=7
  #         #, fontface="italic" 
  #         ) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.16), color = "blue", fontface="bold", label = ("Accuracy in NCs")) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.12), color = "blue", label = paste("italic(r) == ", round(cor(data_non$Age, data_non$Pred_Age_DBN_delange), 2)), parse = TRUE) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.08), color = "blue", label = paste("italic(R) ^ 2 == ", round(cor(data_non$Age, data_non$Pred_Age_DBN_delange)^2, 2)), parse = TRUE) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.04), color = "blue", label = paste("italic(MAE) == ", round(MAE(data_non$Age, data_non$Pred_Age_DBN_delange), 2)), parse = TRUE) +
  annotate("text", x = 76, y = min(yscale)+((max(yscale)-min(yscale))*0.00), color = "blue", label = paste("italic(RMSE) == ", round(RMSE(data_non$Age, data_non$Pred_Age_DBN_delange), 2)), parse = TRUE) +
  theme_bw(base_size = 15) +
theme(legend.position=c(.2, .9), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        , axis.text.x = element_blank()
        , axis.ticks.x = element_blank()
        , axis.text.y = element_blank()
        , axis.ticks.y = element_blank()
        )

scatter_dif_mutationsym_performance_agecorrected

ggsave(paste(out_path, '/dif_mutation/dian_df15_dif_mutationsym_scatter_performance_agecorrected_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

test_r2_non = cor(data_non$Pred_Age_DBN_delange, data_non$Age)^2
test_mae_non = mae(data_non$Pred_Age_DBN_delange, data_non$Age)
test_rmse_non = rmse(data_non$Pred_Age_DBN_delange, data_non$Age)

test_r2_car = cor(data_car$Pred_Age_DBN_delange, data_car$Age)^2
test_mae_car = mae(data_car$Pred_Age_DBN_delange, data_car$Age)
test_rmse_car = rmse(data_car$Pred_Age_DBN_delange, data_car$Age)
```





#Test continuous relationships
##BAG ~ EYO 
###GAMM
```{r}
data_lme$Mutation_Status = ordered(data_lme$Mutation_Status)

eyo_dbn_mut_lme_all_gam_ci = gamm4( #lmer(
  BAG_Choice ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , by = Mutation_Status
        , k = 4, bs = 'cr'
        )
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme
  , REML = TRUE
)
summary(eyo_dbn_mut_lme_all_gam_ci$mer)
summary(eyo_dbn_mut_lme_all_gam_ci$gam)



plot(eyo_dbn_mut_lme_all_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_dbn_mut_lme_all_gam_ci$gam)

concurvity(eyo_dbn_mut_lme_all_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-reviSited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = unique(Mutation_Status)
  , Sex = "Female"#unique(Sex)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
))

Model = eyo_dbn_mut_lme_all_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_dbn_mut_lme_all_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_dbn_mut_lme_all_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

eyo_dbn_mut_lme_all_gam_ci_pred_car = subset(eyo_dbn_mut_lme_all_gam_ci_pred, Mutation_Status == "Carrier")
eyo_dbn_mut_lme_all_gam_ci_pred_non = subset(eyo_dbn_mut_lme_all_gam_ci_pred, Mutation_Status == "Non-Carrier")

eyo_dbn_mut_lme_all_gam_ci_pred_car$dif1 = eyo_dbn_mut_lme_all_gam_ci_pred_car$lwrS - eyo_dbn_mut_lme_all_gam_ci_pred_non$uprS
eyo_dbn_mut_lme_all_gam_ci_pred_sig1 = subset(eyo_dbn_mut_lme_all_gam_ci_pred_car, dif1 > 0)
eyo_dbn_mut_lme_all_gam_ci_pred_sig_point1 = min(eyo_dbn_mut_lme_all_gam_ci_pred_sig1$EYO)

eyo_dbn_mut_lme_all_gam_ci_pred_car$dif2 = eyo_dbn_mut_lme_all_gam_ci_pred_car$fit - eyo_dbn_mut_lme_all_gam_ci_pred_non$uprS
eyo_dbn_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_dbn_mut_lme_all_gam_ci_pred_car, dif2 > 0)
eyo_dbn_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_dbn_mut_lme_all_gam_ci_pred_sig2, EYO > -30)
eyo_dbn_mut_lme_all_gam_ci_pred_sig_point2 = min(eyo_dbn_mut_lme_all_gam_ci_pred_sig2$EYO)


eyo_dbn_mut_lme_all_gam_ci_pred_car$dif3 = eyo_dbn_mut_lme_all_gam_ci_pred_car$lwrS - eyo_dbn_mut_lme_all_gam_ci_pred_non$fit
eyo_dbn_mut_lme_all_gam_ci_pred_sig3 = subset(eyo_dbn_mut_lme_all_gam_ci_pred_car, dif3 > 0)
eyo_dbn_mut_lme_all_gam_ci_pred_sig_point3 = min(eyo_dbn_mut_lme_all_gam_ci_pred_sig3$EYO)

eyo_sig_dif = eyo_dbn_mut_lme_all_gam_ci_pred_sig_point1

cor_eyo_dbn_gam_ci = ggplot(data_lme, aes(x = EYO, y = BAG_Choice, colour = Mutation_Status)) +
  geom_point(aes(color = Mutation_Status, shape = Clin_Status)) +
  labs(title="",x="EYO", y = "BAG (Age-Corrected)") +
  geom_smooth(data = eyo_dbn_mut_lme_all_gam_ci_pred
              , aes(color = Mutation_Status
                    #, fill = Mutation_Status
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_dbn_mut_lme_all_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = Mutation_Status
                    , fill = Mutation_Status)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_dbn_mut_lme_all_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = Mutation_Status)) + 
  scale_color_manual(values = c("blue", "red")) +
  scale_fill_manual(values = c("blue", "red")) +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  geom_hline(yintercept = 0, linetype = "longdash") + 
  geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(round(eyo_sig_dif, 2), 0)
                     , breaks = c(eyo_sig_dif, 0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    

cor_eyo_dbn_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_eyo_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)
```

###Merge Mut Sym Dif Fig-Paper
```{r}
#install.packages()
ggarrange(
  ggarrange(scatter_dif_mutationsym, violin_dif_mutationsym, ncol = 2
            , labels = c("A", "B"))
  , cor_eyo_dbn_gam_ci
  #, bateman_curves
  , labels = c("", "C")
  , nrow = 2
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(ms_path, '/Figure2_MutationDifs_', ver_date, '.png', sep = ""), width = 12, height = 12, units = "in", dpi = 320)
```

###Codon subroups
```{r}

data_lme = subset(data, educ != 'NA')
data_lme = data_lme[order(data_lme$EYO),]
data_lme_car = subset(data_lme, Mutation_Status == "Carrier")
data_lme_car = data_lme_car[order(data_lme_car$EYO),]
data_lme_non = subset(data_lme, Mutation_Status == "Non-Carrier")
data_lme_non = data_lme_non[order(data_lme_non$EYO),]
#table(data$fam_id)
data_lme_codon = subset(data_lme_car, codon_cat != 'PSEN2')
data_lme_codon$codon_cat = ordered(data_lme_codon$codon_cat)

eyo_dbn_codon_lme_gam_ci = gamm4( #lmer(
  BAG_Choice ~ 
    codon_cat
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = codon_cat
        )
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme_codon
  , REML = TRUE
)
#data_lme_codon$codon_cat
summary(eyo_dbn_codon_lme_gam_ci$mer)
summary(eyo_dbn_codon_lme_gam_ci$gam)

plot(eyo_dbn_codon_lme_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_dbn_codon_lme_gam_ci$gam)

concurvity(eyo_dbn_codon_lme_gam_ci$gam, full = TRUE)


#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-reviSited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme_codon, expand.grid(
  EYO = unique(EYO)
  , codon_cat = unique(codon_cat)
  , Sex = "Female"#unique(Sex)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
))

Model = eyo_dbn_codon_lme_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_dbn_codon_lme_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_dbn_codon_lme_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

cor_eyo_dbn_codon_lme_gam_ci = ggplot(data_lme_codon, aes(x = EYO, y = BAG_Choice, colour = codon_cat)) +
  geom_point(aes(color = codon_cat, shape = Clin_Status)) +
  labs(title="Mutation Variants (Carriers Only)",x="EYO", y = "BAG (Age-Corrected)") +
  geom_smooth(data = eyo_dbn_codon_lme_gam_ci_pred
              , aes(color = codon_cat
                    #, fill = codon_cat
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_dbn_codon_lme_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = codon_cat
                    , fill = codon_cat)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_dbn_codon_lme_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = codon_cat)) + 
  #scale_color_manual(values = c("blue", "red")) +
  #scale_fill_manual(values = c("blue", "red")) +
  scale_color_tableau()+#palette = "Tableau 10") +
  scale_fill_tableau()+#palette = "Tableau 10") +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  geom_hline(yintercept = 0, linetype = "longdash") + 
  #geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(0)
                     , breaks = c(0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.12, .75), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    

cor_eyo_dbn_codon_lme_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_codon_eyo_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)
```


###Sex Differences
```{r}

data_lme = subset(data, educ != 'NA')
data_lme = data_lme[order(data_lme$EYO),]
data_lme_car = subset(data_lme, Mutation_Status == "Carrier")
data_lme_car = data_lme_car[order(data_lme_car$EYO),]
data_lme_non = subset(data_lme, Mutation_Status == "Non-Carrier")
data_lme_non = data_lme_non[order(data_lme_non$EYO),]
data_lme_sex = subset(data_lme, Sex != 'NA')
data_lme_sex = subset(data_lme_sex, codon_cat != 'NA')
data_lme_sex$sex_cat = ordered(data_lme_sex$Sex)
table(data_lme_sex$codon_cat)
data_lme_sex_car = subset(data_lme_sex, Mutation_Status == "Carrier")

eyo_dbn_sex_lme_gam_ci = gamm4( #lmer(
  BAG_Choice ~ 
    codon_cat
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = sex_cat
        )
  + sex_cat
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme_sex_car
  , REML = TRUE
)
#data_lme_sex_car$sex_cat
summary(eyo_dbn_sex_lme_gam_ci$mer)
summary(eyo_dbn_sex_lme_gam_ci$gam)

plot(eyo_dbn_sex_lme_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

help(plot.gam)


gam.check(eyo_dbn_sex_lme_gam_ci$gam)

concurvity(eyo_dbn_sex_lme_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-reviSited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme_sex_car, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = "Carrier"
  , codon_cat = "PSEN1 Codon 200+"
  , sex_cat = unique(sex_cat)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
))

Model = eyo_dbn_sex_lme_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_dbn_sex_lme_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_dbn_sex_lme_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

cor_eyo_dbn_sex_lme_gam_ci = ggplot(data_lme_sex_car, aes(x = EYO, y = BAG_Choice, colour = sex_cat)) +
  geom_point(aes(color = sex_cat, shape = Clin_Status)) +
  labs(title="Sex Differences (Carriers Only)",x="EYO", y = "BAG (Age-Corrected)") +
  geom_smooth(data = eyo_dbn_sex_lme_gam_ci_pred
              , aes(color = sex_cat
                    #, fill = sex_cat
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_dbn_sex_lme_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = sex_cat
                    , fill = sex_cat)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_dbn_sex_lme_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = sex_cat)) + 
  #scale_color_manual(values = c("blue", "red")) +
  #scale_fill_manual(values = c("blue", "red")) +
  scale_color_tableau()+#palette = "Tableau 10") +
  scale_fill_tableau()+#palette = "Tableau 10") +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  geom_hline(yintercept = 0, linetype = "longdash") + 
  #geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(0)
                     , breaks = c(0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    
cor_eyo_dbn_sex_lme_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_sex_eyo_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)


compare_means(BAG_Choice ~ sex_cat,  data = data_lme_sex, method = "wilcox.test")
my_comparisons <- list(  c("Female", "Male") )

violin_dif_sex = ggplot(data_lme_sex, aes(x=Sex, y=BAG_Choice, fill=Sex)) +
  labs(title="Sex",x="Sex", y = "BAG (Age-Corrected)") +
  geom_violin(trim=FALSE, color = "black") + 
  #scale_fill_manual(breaks=c("Non-Carrier", "Asymptomatic MC", "Symptomatic MC"),
  #                            values=c("blue", "orange", "red")) +
  #scale_fill_viridis(discrete = TRUE) + 
  scale_fill_tableau()+
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot( width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
                     , aes(label = ..p.adj..)
                     )+ # Add pairwise comparisons p-value
  #stat_compare_means(label.y = 50)+     # Add global p-value
  facet_wrap(~Mutation_Status)+
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5))

violin_dif_sex

ggsave(paste(out_path, '/cor_eyo/dian_df15_dif_sex_violin_', ver_date, '.png', sep = ""), width = 5, height = 5, units = "in", dpi = 320)
```

###APOE
```{r}

data_lme = subset(data, educ != 'NA')
data_lme = data_lme[order(data_lme$EYO),]
data_lme_car = subset(data_lme, Mutation_Status == "Carrier")
data_lme_car = data_lme_car[order(data_lme_car$EYO),]
data_lme_non = subset(data_lme, Mutation_Status == "Non-Carrier")
data_lme_non = data_lme_non[order(data_lme_non$EYO),]
data_lme_apoe = subset(data_lme, apoe != '.')
data_lme_apoe = subset(data_lme_apoe, codon_cat != 'NA')
data_lme_apoe$apoe_cat = ordered(data_lme_apoe$apoe_pos)
table(data_lme_apoe$apoe_cat)

data_lme_apoe_car = subset(data_lme_apoe, Mutation_Status == "Carrier")

eyo_dbn_apoe_lme_gam_ci = gamm4( #lmer(
  BAG_Choice ~ 
    codon_cat
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = apoe_cat
        )  
  + Sex
  + educ
  + apoe_cat
  , random = ~ (1 | fam_id)
  , data = data_lme_apoe_car
  , REML = TRUE
)
#data_lme_apoe_car$apoe_cat
summary(eyo_dbn_apoe_lme_gam_ci$mer)
summary(eyo_dbn_apoe_lme_gam_ci$gam)

plot(eyo_dbn_apoe_lme_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_dbn_apoe_lme_gam_ci$gam)

concurvity(eyo_dbn_apoe_lme_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-reviSited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme_apoe_car, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = "Carrier"
  , codon_cat = "PSEN1 Codon 200+"
  , Sex = "Female"
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_cat = unique(apoe_cat)
  #, fam_id = unique(fam_id)
))

Model = eyo_dbn_apoe_lme_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_dbn_apoe_lme_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_dbn_apoe_lme_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

cor_eyo_dbn_apoe_lme_gam_ci = ggplot(data_lme_apoe_car, aes(x = EYO, y = BAG_Choice, colour = apoe_cat)) +
  geom_point(aes(color = apoe_cat, shape = Clin_Status)) +
  labs(title="APOE Differences (Carriers Only)",x="EYO", y = "BAG (Age-Corrected)") +
  geom_smooth(data = eyo_dbn_apoe_lme_gam_ci_pred
              , aes(color = apoe_cat
                    #, fill = apoe_cat
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_dbn_apoe_lme_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = apoe_cat
                    , fill = apoe_cat)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_dbn_apoe_lme_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = apoe_cat)) + 
  #scale_color_manual(values = c("blue", "red")) +
  #scale_fill_manual(values = c("blue", "red")) +
  scale_color_tableau()+#palette = "Tableau 10") +
  scale_fill_tableau()+#palette = "Tableau 10") +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  geom_hline(yintercept = 0, linetype = "longdash") + 
  #geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(0)
                     , breaks = c(0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    
cor_eyo_dbn_apoe_lme_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_apoe_eyo_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)

data_lme_apoe = data_lme_apoe%>%
  mutate(mut_apoe = case_when(
    Mutation_Status == "Non-Carrier" & apoe_cat == "E4-" ~ "Non-Carrier / E4-",
    Mutation_Status == "Non-Carrier" & apoe_cat == "E4+" ~ "Non-Carrier / E4+",
    Mutation_Status == "Carrier" & apoe_cat == "E4-" ~ "Carrier / E4-",
    Mutation_Status == "Carrier" & apoe_cat == "E4+" ~ "Carrier / E4+"))

compare_means(BAG_Choice ~ apoe_cat,  data = data_lme_apoe, method = "wilcox.test")
my_comparisons <- list(  c("E4-", "E4+") )

#compare_means(BAG_Choice ~ mut_apoe,  data = data_lme_apoe, method = "wilcox.test")
#my_comparisons <- list(  c("E4-", "E4+") )

violin_dif_apoe = ggplot(data_lme_apoe, aes(x=apoe_cat, y=BAG_Choice, fill=apoe_cat)) +
  labs(title="APOE",x="APOE", y = "BAG (Age-Corrected)") +
  
  geom_violin(trim=FALSE, color = "black"
                #, draw_quantiles = c(0.25, 0.5, 0.75)
              ) + 
  #scale_fill_manual(breaks=c("Non-Carrier", "Asymptomatic MC", "Symptomatic MC"),
  #                            values=c("blue", "orange", "red")) +
  #scale_fill_viridis(discrete = TRUE) + 
  scale_fill_tableau()+
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, 
               color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
                     , aes(label = ..p.adj..)
                     )+ # Add pairwise comparisons p-value
  #stat_compare_means(label.y = 50)+     # Add global p-value
  facet_wrap(~Mutation_Status) +
  theme_classic(base_size = 15) +
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5))

violin_dif_apoe

ggsave(paste(out_path, '/cor_eyo/dian_df15_dif_apoe_violin_', ver_date, '.png', sep = ""), width = 5, height = 5, units = "in", dpi = 320)
```
###Education
```{r}

data_lme = subset(data, educ != 'NA')
data_lme = data_lme[order(data_lme$EYO),]
data_lme_car = subset(data_lme, Mutation_Status == "Carrier")
data_lme_car = data_lme_car[order(data_lme_car$EYO),]
data_lme_non = subset(data_lme, Mutation_Status == "Non-Carrier")
data_lme_non = data_lme_non[order(data_lme_non$EYO),]
data_lme_edu = subset(data_lme, educ < 27)
data_lme_edu = subset(data_lme_edu, codon_cat != 'NA')
data_lme_edu$edu_cat = ordered(data_lme_edu$edu_med)
table(data_lme_edu$edu_cat)

eyo_dbn_edu_lme_gam_ci = gamm4( #lmer(
  BAG_Choice ~ 
    codon_cat
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = edu_cat
        )   
  + Mutation_Status*educ
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme_edu
  , REML = TRUE
)
#data_lme_edu$edu_cat
summary(eyo_dbn_edu_lme_gam_ci$mer)
summary(eyo_dbn_edu_lme_gam_ci$gam)

plot(eyo_dbn_edu_lme_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_dbn_edu_lme_gam_ci$gam)

concurvity(eyo_dbn_edu_lme_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-reviSited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme_edu, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = "Carrier"
  , codon_cat = "PSEN1 Codon 200+"
  , Sex = "Female"
  , edu_cat = unique(edu_cat)
  , educ = median(data_lme_edu$educ)
  , apoe_pos = "E4-"
  #, fam_id = unique(fam_id)
))

Model = eyo_dbn_edu_lme_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_dbn_edu_lme_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_dbn_edu_lme_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )



cor_eyo_dbn_edu_lme_gam_ci = ggplot(data_lme_edu, aes(x = EYO, y = BAG_Choice, colour = edu_cat)) +
  geom_point(aes(color = edu_cat, shape = Clin_Status)) +
  labs(title="",x="EYO", y = "BAG (Age-Corrected)") +
  geom_smooth(data = eyo_dbn_edu_lme_gam_ci_pred
              , aes(color = edu_cat
                    #, fill = edu_cat
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_dbn_edu_lme_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = edu_cat
                    , fill = edu_cat)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_dbn_edu_lme_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = edu_cat)) + 
  #scale_color_manual(values = c("blue", "red")) +
  #scale_fill_manual(values = c("blue", "red")) +
  scale_color_tableau()+#palette = "Tableau 10") +
  scale_fill_tableau()+#palette = "Tableau 10") +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  geom_hline(yintercept = 0, linetype = "longdash") + 
  #geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(0)
                     , breaks = c(0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    
cor_eyo_dbn_edu_lme_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_edu_eyo_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)

#data_lme_edu = data_lme_edu%>%
#  mutate(mut_edu = case_when(
#    Mutation_Status == "Non-Carrier" & edu_cat == "E4-" ~ "Non-Carrier / E4-",
#    Mutation_Status == "Non-Carrier" & edu_cat == "E4+" ~ "Non-Carrier / E4+",
#    Mutation_Status == "Carrier" & edu_cat == "E4-" ~ "Carrier / E4-",
#    Mutation_Status == "Carrier" & edu_cat == "E4+" ~ "Carrier / E4+"))

compare_means(BAG_Choice ~ edu_cat,  data = data_lme_edu, method = "wilcox.test")
my_comparisons <- list(  c("<= Median Education", "> Median Education") )

#compare_means(BAG_Choice ~ mut_edu,  data = data_lme_edu, method = "wilcox.test")
#my_comparisons <- list(  c("E4-", "E4+") )

violin_dif_edu = ggplot(data_lme_edu, aes(x=edu_cat, y=BAG_Choice, fill=edu_cat)) +
  labs(title="Education",x="", y = "BAG (Age-Corrected)") +
  
  geom_violin(trim=FALSE, color = "black"
                #, draw_quantiles = c(0.25, 0.5, 0.75)
              ) + 
  #scale_fill_manual(breaks=c("Non-Carrier", "Asymptomatic MC", "Symptomatic MC"),
  #                            values=c("blue", "orange", "red")) +
  #scale_fill_viridis(discrete = TRUE) + 
  scale_fill_tableau()+
  geom_hline(yintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, 
               color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
                     , aes(label = ..p.adj..)
                     )+ # Add pairwise comparisons p-value
  #stat_compare_means(label.y = 50)+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5))

violin_dif_edu

ggsave(paste(out_path, '/cor_eyo/dian_df15_dif_edu_violin_', ver_date, '.png', sep = ""), width = 5, height = 5, units = "in", dpi = 320)

grouped_cor_edu = ggplot(data_lme_edu, aes(y = BAG_Choice, x = educ)) +
  geom_point(aes(color = Mutation_Status)) +
  labs(title="Education", y = "BAG (Age-Corrected)", x = "Education (years)") +
  geom_smooth(aes(color = Mutation_Status, fill = Mutation_Status), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(Mutation_Status = imagid.x, color = Mutation_Status)) + 
  scale_color_manual(values = c("blue",  "red", "black")
                     , labels = c('Non-Carrier', 'Carrier', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "red", "grey"), labels = c('Non-Carrier', 'Carrier', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = Mutation_Status)
           , label.y = c(34, 30)
           , label.x = c(15, 15)
           ) +
  theme_bw(base_size = 15) +
  xlim(8.5, 25) +
  theme(legend.position='none', plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

grouped_cor_edu

ggsave(paste(out_path, '/cor_eyo/dian_df15_grouped_cor_BAG_Choice_edu_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

data_lme_edu$whole_sample = 'whole_sample'

ungrouped_cor_edu = ggplot(data_lme_edu, aes(y = BAG_Choice, x = educ, colour = whole_sample)) +
  geom_point() +
  labs(title="Education", y = "BAG (Age-Corrected)", x = "Education (years)") +
  geom_smooth(aes(fill = whole_sample, colour = whole_sample), method = lm) + 
  #geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  #geom_line(aes()) + 
  scale_color_tableau() +
  scale_fill_tableau() +
  #scale_color_manual(values = c("blue", "orange", "red")) +
  #scale_fill_manual(values = c("blue", "orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(label.y = c(-28)
           , label.x = c(12.5)) +
  theme_bw(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

ungrouped_cor_edu

ggsave(paste(out_path, '/cor_eyo/dian_df15_ungrouped_cor_BAG_Choice_edu_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```


###Full GAMM
```{r}
data_lme$Mutation_Status = ordered(data_lme$Mutation_Status)

eyo_dbn_mut_lme_all_gam_ci_het = gamm4( #lmer(
  BAG_Choice ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , by = Mutation_Status
        , k = 4, bs = 'cr'
        )
  + codon_cat#*Mutation_Status
  + Sex#*Mutation_Status
  + educ#*Mutation_Status
  + apoe_pos*Mutation_Status
  , random = ~ (1 | fam_id)
  , data = data_lme
  , REML = TRUE
)
summary(eyo_dbn_mut_lme_all_gam_ci_het$mer)
summary(eyo_dbn_mut_lme_all_gam_ci_het$gam)
#coef(eyo_dbn_mut_lme_all_gam_ci_het$gam)
#summary(eyo_dbn_mut_lme_all_gam_ci_het$gam)$p.table


plot(eyo_dbn_mut_lme_all_gam_ci_het$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_dbn_mut_lme_all_gam_ci_het$gam)

concurvity(eyo_dbn_mut_lme_all_gam_ci_het$gam, full = TRUE)
```

###Merge heterogeneity fig Paper
```{r}
ggarrange(
  cor_eyo_dbn_codon_lme_gam_ci, 
  ggarrange(violin_dif_apoe, violin_dif_sex, grouped_cor_edu
            , ncol = 3
            , labels = c("B", "C", "D"))
  , labels = c("A", "")
  , nrow = 2
  , heights = c(1, .75)
  #, common.legend = TRUE, legend = "bottom"
)
ggsave(paste(ms_path, '/Figure5_Heterogeneity_', ver_date, '.png', sep = ""), width = 10, height = 8, units = "in", dpi = 320)
```
###Merge supplemental heterogeneity fig Paper
```{r}
ggarrange(
  cor_eyo_dbn_apoe_lme_gam_ci, 
  cor_eyo_dbn_sex_lme_gam_ci
  , labels = c("A", "B")
  , nrow = 2
  #, common.legend = TRUE, legend = "bottom"
)
ggsave(paste(ms_path, '/FigureS6_HeterogeneityGAMM_', ver_date, '.png', sep = ""), width = 10, height = 10, units = "in", dpi = 320)
```


##Amyloid
###PIB
```{r}
lme_dbn_PIB_main = lmer(
  BAG_Choice ~ 
    PIB_fSUVR_rsf_TOT_CORTMEAN.x
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_PIB_main)

lme_dbn_PIB_main_anova = anova(lme_dbn_PIB_main)
F_to_eta2(lme_dbn_PIB_main_anova$`F value`[1]
          , df = lme_dbn_PIB_main_anova$NumDF[1]
          , df_error = lme_dbn_PIB_main_anova$DenDF[1])

parameters::model_parameters(lme_dbn_PIB_main, effects = "fixed", ci_method = "satterthwaite")
t_to_eta2(8.47 , df_error = 215.05)

lme_dbn_PIB_interaction = lmer(
  BAG_Choice ~ 
    PIB_fSUVR_rsf_TOT_CORTMEAN.x*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_PIB_interaction)

lme_dbn_PIB_interaction_anova = anova(lme_dbn_PIB_interaction)
F_to_eta2(lme_dbn_PIB_interaction_anova$`F value`[1]
          , df = lme_dbn_PIB_interaction_anova$NumDF[1]
          , df_error = lme_dbn_PIB_interaction_anova$DenDF[1])
F_to_eta2(lme_dbn_PIB_interaction_anova$`F value`[6]
          , df = lme_dbn_PIB_interaction_anova$NumDF[6]
          , df_error = lme_dbn_PIB_interaction_anova$DenDF[6])
#F_to_eta2(6.2416 , 1, 226.90)

cor_pib = ggplot(data_car, aes(y = BAG_Choice, x = PIB_fSUVR_rsf_TOT_CORTMEAN.x)) +
  geom_point(aes(color = group)) +
  labs(title="PIB Cortical Mean (RSF, SUVR)", y = "BAG (Age-Corrected)", x = "PIB (SUVR)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 1.42, linetype = "dotted") +
  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  stat_cor(aes(color = group), label.x = 3.5, label.y = c(-15,-18)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_pib

ggsave(paste(out_path, '/cor_A/dian_df15_cor_BAG_Choice_PIB_SUVR_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```
###PIB_log
```{r}
hist(data_car$PIB_fSUVR_rsf_TOT_CORTMEAN.x, breaks = 50)
data_car$PIB_log_fSUVR_rsf_TOT_CORTMEAN.x = log(data_car$PIB_fSUVR_rsf_TOT_CORTMEAN.x)
hist(data_car$PIB_log_fSUVR_rsf_TOT_CORTMEAN.x, breaks = 50)
```

```{r}
lme_dbn_PIB_log_main = lmer(
  BAG_Choice ~ 
    PIB_log_fSUVR_rsf_TOT_CORTMEAN.x
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_PIB_log_main)

lme_dbn_PIB_log_main_anova = anova(lme_dbn_PIB_log_main)
F_to_eta2(lme_dbn_PIB_log_main_anova$`F value`[1]
          , df = lme_dbn_PIB_log_main_anova$NumDF[1]
          , df_error = lme_dbn_PIB_log_main_anova$DenDF[1])


lme_dbn_PIB_log_interaction = lmer(
  BAG_Choice ~ 
    PIB_log_fSUVR_rsf_TOT_CORTMEAN.x*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_PIB_log_interaction)

lme_dbn_PIB_log_interaction_anova = anova(lme_dbn_PIB_log_interaction)
F_to_eta2(lme_dbn_PIB_log_interaction_anova$`F value`[6]
          , df = lme_dbn_PIB_log_interaction_anova$NumDF[6]
          , df_error = lme_dbn_PIB_log_interaction_anova$DenDF[6])
#F_to_eta2(6.2416 , 1, 226.90)

cor_PIB_log = ggplot(data_car, aes(y = BAG_Choice, x = PIB_log_fSUVR_rsf_TOT_CORTMEAN.x)) +
  geom_point(aes(color = group)) +
  labs(title="PIB Cortical Mean (log)", y = "BAG (Age-Corrected)", x = "PIB (log)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #geom_vline(xintercept = 1.42, linetype = "dotted") +
  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  stat_cor(aes(color = group)#, label.x = 4, label.y = c(-24,-27)
           ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_PIB_log

ggsave(paste(out_path, '/cor_A/dian_df15_cor_BAG_Choice_PIB_log_SUVR_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```


###AB42/40
```{r}
lme_dbn_CSF_AB4240_ratio_main = lmer(
  BAG_Choice ~ 
    CSF_AB4240_ratio
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_AB4240_ratio_main)

lme_dbn_CSF_AB4240_ratio_main_anova = anova(lme_dbn_CSF_AB4240_ratio_main)
F_to_eta2(lme_dbn_CSF_AB4240_ratio_main_anova$`F value`[1]
          , df = lme_dbn_CSF_AB4240_ratio_main_anova$NumDF[1]
          , df_error = lme_dbn_CSF_AB4240_ratio_main_anova$DenDF[1])
#F_to_eta2(7.2314 , 1, 216.43  )


lme_dbn_CSF_AB4240_ratio_interaction = lmer(
  BAG_Choice ~ 
    CSF_AB4240_ratio*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_AB4240_ratio_interaction)

cor_ab4240 = ggplot(data_car, aes(y = BAG_Choice, x = CSF_AB4240_ratio)) +
  geom_point(aes(color = group)) +
  labs(title="CSF AB42/40 Ratio", y = "BAG (Age-Corrected)", x = "CSF AB42/40") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0.0673, linetype = "dotted") +
  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  stat_cor(aes(color = group), label.x = .125, label.y = c(-15,-18)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.8, .875), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_ab4240

ggsave(paste(out_path, '/cor_A/dian_df15_cor_BAG_Choice_CSF_AB4240_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```
###AB42/40_log
```{r}
hist(data_car$CSF_AB4240_ratio, breaks = 50)
data_car$CSF_AB4240_log_ratio = log(data_car$CSF_AB4240_ratio)
hist(data_car$CSF_AB4240_log_ratio, breaks = 50)
```

```{r}
lme_dbn_CSF_AB4240_log_ratio_main = lmer(
  BAG_Choice ~ 
    CSF_AB4240_log_ratio
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_AB4240_log_ratio_main)

lme_dbn_CSF_AB4240_log_ratio_main_anova = anova(lme_dbn_CSF_AB4240_log_ratio_main)
F_to_eta2(lme_dbn_CSF_AB4240_log_ratio_main_anova$`F value`[1]
          , df = lme_dbn_CSF_AB4240_log_ratio_main_anova$NumDF[1]
          , df_error = lme_dbn_CSF_AB4240_log_ratio_main_anova$DenDF[1])
#F_to_eta2(7.2314 , 1, 216.43  )


lme_dbn_CSF_AB4240_log_ratio_interaction = lmer(
  BAG_Choice ~ 
    CSF_AB4240_log_ratio*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_AB4240_log_ratio_interaction)

cor_ab4240_log = ggplot(data_car, aes(y = BAG_Choice, x = CSF_AB4240_log_ratio)) +
  geom_point(aes(color = group)) +
  labs(title="CSF AB42/40 Ratio (log)", y = "BAG (Age-Corrected)", x = "CSF AB42/40 (log)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #geom_vline(xintercept = 0.0673, linetype = "dotted") +
  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  stat_cor(aes(color = group)#, label.x = .125, label.y = c(-24,-27)
           ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.8, .875), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_ab4240_log

ggsave(paste(out_path, '/cor_A/dian_df15_cor_BAG_Choice_CSF_AB4240_log_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```


###Merge A Fig
```{r}
ggarrange(
  cor_pib, cor_ab4240
  
  #, labels = c("A", "B", "C", "D", "E", "F", "G", "H")
  , nrow = 1, ncol = 2
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(out_path, '/poster_figs/dian_df15_BAG_Choice_A_Figure_', ver_date, '.png', sep = ""), width = 10, height = 5, units = "in", dpi = 320)
```
##Tau

###Plasma pTau
```{r}
lme_dbn_pTau_Plasma_main = lmer(
  BAG_Choice ~ 
    pTau_Plasma
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_pTau_Plasma_main)

lme_dbn_pTau_Plasma_main_anova = anova(lme_dbn_pTau_Plasma_main)
F_to_eta2(lme_dbn_pTau_Plasma_main_anova$`F value`[1]
          , df = lme_dbn_pTau_Plasma_main_anova$NumDF[1]
          , df_error = lme_dbn_pTau_Plasma_main_anova$DenDF[1])
#F_to_eta2(7.2314 , 1, 216.43  )

lme_dbn_pTau_Plasma_interaction = lmer(
  BAG_Choice ~ 
    pTau_Plasma*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_pTau_Plasma_interaction)

lme_dbn_pTau_Plasma_interaction_anova = anova(lme_dbn_pTau_Plasma_interaction)
F_to_eta2(lme_dbn_pTau_Plasma_interaction_anova$`F value`[1]
          , df = lme_dbn_pTau_Plasma_interaction_anova$NumDF[1]
          , df_error = lme_dbn_pTau_Plasma_interaction_anova$DenDF[1])

cor_ptau_plasma = ggplot(data_car, aes(y = BAG_Choice, x = pTau_Plasma))+
  geom_point(aes(color = group)) +
  labs(title="Plasma pTau-181", y = "BAG (Age-Corrected)", x = "Plasma pTau (pg/mL)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_ptau_plasma

ggsave(paste(out_path, '/cor_T/dian_df15_cor_BAG_Choice_Plasma_pTau_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```
###Plasma pTau_log

```{r}
hist(data_car$pTau_Plasma, breaks = 50)
data_car$pTau_log_Plasma = log(data_car$pTau_Plasma)
hist(data_car$pTau_log_Plasma, breaks = 50)
```

```{r}
lme_dbn_pTau_log_Plasma_main = lmer(
  BAG_Choice ~ 
    pTau_log_Plasma
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_pTau_log_Plasma_main)

lme_dbn_pTau_log_Plasma_main_anova = anova(lme_dbn_pTau_log_Plasma_main)
F_to_eta2(lme_dbn_pTau_log_Plasma_main_anova$`F value`[1]
          , df = lme_dbn_pTau_log_Plasma_main_anova$NumDF[1]
          , df_error = lme_dbn_pTau_log_Plasma_main_anova$DenDF[1])
#F_to_eta2(7.2314 , 1, 216.43  )

lme_dbn_pTau_log_Plasma_interaction = lmer(
  BAG_Choice ~ 
    pTau_log_Plasma*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_pTau_log_Plasma_interaction)

cor_pTau_log_plasma = ggplot(data_car, aes(y = BAG_Choice, x = pTau_log_Plasma))+
  geom_point(aes(color = group)) +
  labs(title="Plasma pTau-181 (log)", y = "BAG (Age-Corrected)", x = "Plasma pTau (log)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_pTau_log_plasma

ggsave(paste(out_path, '/cor_T/dian_df15_cor_BAG_Choice_Plasma_pTau_log_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```

###CSF pTau/ab40
```{r}
lme_dbn_CSF_ptauab40_ratio_main = lmer(
  BAG_Choice ~ 
    CSF_ptauab40_ratio
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_ptauab40_ratio_main)

lme_dbn_CSF_ptauab40_ratio_main_anova = anova(lme_dbn_CSF_ptauab40_ratio_main)
F_to_eta2(lme_dbn_CSF_ptauab40_ratio_main_anova$`F value`[1]
          , df = lme_dbn_CSF_ptauab40_ratio_main_anova$NumDF[1]
          , df_error = lme_dbn_CSF_ptauab40_ratio_main_anova$DenDF[1])
#F_to_eta2(7.2314 , 1, 216.43  )

lme_dbn_CSF_ptauab40_ratio_interaction = lmer(
  BAG_Choice ~ 
    CSF_ptauab40_ratio*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_ptauab40_ratio_interaction)
lme_dbn_CSF_ptauab40_ratio_interaction_anova = anova(lme_dbn_CSF_ptauab40_ratio_interaction)
F_to_eta2(lme_dbn_CSF_ptauab40_ratio_interaction_anova$`F value`[1]
          , df = lme_dbn_CSF_ptauab40_ratio_interaction_anova$NumDF[1]
          , df_error = lme_dbn_CSF_ptauab40_ratio_interaction_anova$DenDF[1])

F_to_eta2(lme_dbn_CSF_ptauab40_ratio_interaction_anova$`F value`[6]
          , df = lme_dbn_CSF_ptauab40_ratio_interaction_anova$NumDF[6]
          , df_error = lme_dbn_CSF_ptauab40_ratio_interaction_anova$DenDF[6])
#F_to_eta2(7.2314 , 1, 216.43  )

cor_ptauab40_csf = ggplot(data_car, aes(y = BAG_Choice, x = CSF_ptauab40_ratio)) +
  geom_point(aes(color = group)) +
  labs(title="CSF pTau/AB40 Ratio", y = "BAG (Age-Corrected)", x = "CSF pTau/AB40") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.8, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_ptauab40_csf

ggsave(paste(out_path, '/cor_T/dian_df15_cor_BAG_Choice_CSF_pTauAB40_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```
###CSF pTau/ab40_log
```{r}
hist(data_car$CSF_ptauab40_ratio, breaks = 50)
data_car$CSF_ptauab40_log_ratio = log(data_car$CSF_ptauab40_ratio)
hist(data_car$CSF_ptauab40_log_ratio, breaks = 50)
```

```{r}
lme_dbn_CSF_ptauab40_log_ratio_main = lmer(
  BAG_Choice ~ 
    CSF_ptauab40_log_ratio
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_ptauab40_log_ratio_main)

lme_dbn_CSF_ptauab40_log_ratio_main_anova = anova(lme_dbn_CSF_ptauab40_log_ratio_main)
F_to_eta2(lme_dbn_CSF_ptauab40_log_ratio_main_anova$`F value`[1]
          , df = lme_dbn_CSF_ptauab40_log_ratio_main_anova$NumDF[1]
          , df_error = lme_dbn_CSF_ptauab40_log_ratio_main_anova$DenDF[1])
#F_to_eta2(7.2314 , 1, 216.43  )

lme_dbn_CSF_ptauab40_log_ratio_interaction = lmer(
  BAG_Choice ~ 
    CSF_ptauab40_log_ratio*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CSF_ptauab40_log_ratio_interaction)
lme_dbn_CSF_ptauab40_log_ratio_interaction_anova = anova(lme_dbn_CSF_ptauab40_log_ratio_interaction)
F_to_eta2(lme_dbn_CSF_ptauab40_log_ratio_interaction_anova$`F value`[6]
          , df = lme_dbn_CSF_ptauab40_log_ratio_interaction_anova$NumDF[6]
          , df_error = lme_dbn_CSF_ptauab40_log_ratio_interaction_anova$DenDF[6])
#F_to_eta2(7.2314 , 1, 216.43  )

cor_ptauab40_log_csf = ggplot(data_car, aes(y = BAG_Choice, x = CSF_ptauab40_log_ratio)) +
  geom_point(aes(color = group)) +
  labs(title="CSF pTau/AB40 Ratio (log)", y = "BAG (Age-Corrected)", x = "CSF pTau/AB40 (log)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.8, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_ptauab40_log_csf

ggsave(paste(out_path, '/cor_T/dian_df15_cor_BAG_Choice_CSF_pTauab40_log_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```


###Merge T Fig
```{r}
ggarrange(
 cor_ptauab40_csf, cor_ptau_plasma

  #, labels = c("A", "B", "C", "D", "E", "F", "G", "H")
  , nrow = 1, ncol = 2
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(out_path, '/poster_figs/dian_df15_BAG_Choice_T_Figure_', ver_date, '.png', sep = ""), width = 10, height = 5, units = "in", dpi = 320)
```
##Neurodegeneration

###CSF NFL
```{r}
lme_dbn_NFL_CSF_main = lmer(
  BAG_Choice ~ 
    NFL_CSF
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_CSF_main)

lme_dbn_NFL_CSF_main_anova = anova(lme_dbn_NFL_CSF_main)
F_to_eta2(lme_dbn_NFL_CSF_main_anova$`F value`[1]
          , df = lme_dbn_NFL_CSF_main_anova$NumDF[1]
          , df_error = lme_dbn_NFL_CSF_main_anova$DenDF[1])

lme_dbn_NFL_CSF_interaction = lmer(
  BAG_Choice ~ 
    NFL_CSF*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_CSF_interaction)

lme_dbn_NFL_CSF_interaction_anova = anova(lme_dbn_NFL_CSF_interaction)
F_to_eta2(lme_dbn_NFL_CSF_interaction_anova$`F value`[1]
          , df = lme_dbn_NFL_CSF_interaction_anova$NumDF[1]
          , df_error = lme_dbn_NFL_CSF_interaction_anova$DenDF[1])

F_to_eta2(lme_dbn_NFL_CSF_interaction_anova$`F value`[6]
          , df = lme_dbn_NFL_CSF_interaction_anova$NumDF[6]
          , df_error = lme_dbn_NFL_CSF_interaction_anova$DenDF[6])

cor_nfl_csf = ggplot(data_car, aes(y = BAG_Choice, x = NFL_CSF)) +
  geom_point(aes(color = group)) +
  labs(title="CSF NfL", y = "BAG (Age-Corrected)", x = "CSF NfL (pg/mL)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_nfl_csf

ggsave(paste(out_path, '/cor_N/dian_df15_cor_BAG_Choice_NFLCSF_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```

```{r}
lme_dbn_NFL_CSF_main = lmer(
  BAG_Choice ~ 
    NFL_CSF
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data
)
summary(lme_dbn_NFL_CSF_main)

lme_dbn_NFL_CSF_main_anova = anova(lme_dbn_NFL_CSF_main)
F_to_eta2(lme_dbn_NFL_CSF_main_anova$`F value`[1]
          , df = lme_dbn_NFL_CSF_main_anova$NumDF[1]
          , df_error = lme_dbn_NFL_CSF_main_anova$DenDF[1])

lme_dbn_NFL_CSF_interaction = lmer(
  BAG_Choice ~ 
    NFL_CSF*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data
)
summary(lme_dbn_NFL_CSF_interaction)

lme_dbn_NFL_CSF_interaction_anova = anova(lme_dbn_NFL_CSF_interaction)
F_to_eta2(lme_dbn_NFL_CSF_interaction_anova$`F value`[6]
          , df = lme_dbn_NFL_CSF_interaction_anova$NumDF[6]
          , df_error = lme_dbn_NFL_CSF_interaction_anova$DenDF[6])

cor_nfl_csf_3group = ggplot(data, aes(y = BAG_Choice, x = NFL_CSF)) +
  geom_point(aes(color = group)) +
  labs(title="CSF NfL", y = "BAG (Age-Corrected)", x = "CSF NfL (pg/mL)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("blue", "orange", "red")) +
  scale_fill_manual(values = c("blue", "orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_nfl_csf_3group

ggsave(paste(out_path, '/cor_N/dian_df15_cor_BAG_Choice_NFLCSF_nc_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```


###CSF NFL_log

```{r}
hist(data_car$NFL_CSF, breaks = 50)
data_car$NFL_log_CSF = log(data_car$NFL_CSF)
hist(data_car$NFL_log_CSF, breaks = 50)
```

```{r}
lme_dbn_NFL_log_CSF_main = lmer(
  BAG_Choice ~ 
    NFL_log_CSF
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_log_CSF_main)

lme_dbn_NFL_log_CSF_main_anova = anova(lme_dbn_NFL_log_CSF_main)
F_to_eta2(lme_dbn_NFL_log_CSF_main_anova$`F value`[1]
          , df = lme_dbn_NFL_log_CSF_main_anova$NumDF[1]
          , df_error = lme_dbn_NFL_log_CSF_main_anova$DenDF[1])

lme_dbn_NFL_log_CSF_interaction = lmer(
  BAG_Choice ~ 
    NFL_log_CSF*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_log_CSF_interaction)

lme_dbn_NFL_log_CSF_interaction_anova = anova(lme_dbn_NFL_log_CSF_interaction)
F_to_eta2(lme_dbn_NFL_log_CSF_interaction_anova$`F value`[6]
          , df = lme_dbn_NFL_log_CSF_interaction_anova$NumDF[6]
          , df_error = lme_dbn_NFL_log_CSF_interaction_anova$DenDF[6])

cor_NFL_log_csf = ggplot(data_car, aes(y = BAG_Choice, x = NFL_log_CSF)) +
  geom_point(aes(color = group)) +
  labs(title="CSF NfL (log)", y = "BAG (Age-Corrected)", x = "CSF NfL (log)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_NFL_log_csf

ggsave(paste(out_path, '/cor_N/dian_df15_cor_BAG_Choice_NFL_logCSF_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```

###Plasma NFL
```{r}
lme_dbn_NFL_Plasma_main = lmer(
  BAG_Choice ~ 
    NFL_Plasma
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_Plasma_main)


lme_dbn_NFL_Plasma_main_anova = anova(lme_dbn_NFL_Plasma_main)
F_to_eta2(lme_dbn_NFL_Plasma_main_anova$`F value`[1]
          , df = lme_dbn_NFL_Plasma_main_anova$NumDF[1]
          , df_error = lme_dbn_NFL_Plasma_main_anova$DenDF[1])

lme_dbn_NFL_Plasma_interaction = lmer(
  BAG_Choice ~ 
    NFL_Plasma*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_Plasma_interaction)

lme_dbn_NFL_Plasma_interaction_anova = anova(lme_dbn_NFL_Plasma_interaction)
F_to_eta2(lme_dbn_NFL_Plasma_interaction_anova$`F value`[1]
          , df = lme_dbn_NFL_Plasma_interaction_anova$NumDF[1]
          , df_error = lme_dbn_NFL_Plasma_interaction_anova$DenDF[1])

F_to_eta2(lme_dbn_NFL_Plasma_interaction_anova$`F value`[6]
          , df = lme_dbn_NFL_Plasma_interaction_anova$NumDF[6]
          , df_error = lme_dbn_NFL_Plasma_interaction_anova$DenDF[6])

cor_nfl_plasma = ggplot(data_car, aes(y = BAG_Choice, x = NFL_Plasma)) +
  geom_point(aes(color = group)) +
  labs(title="Plasma NfL", y = "BAG (Age-Corrected)", x = "Plasma NfL (pg/mL)") +
  geom_smooth(aes(color = group, fill = group), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c("orange", "red")) +
  scale_fill_manual(values = c("orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group), label.y = c(39.5, 35.5)) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_nfl_plasma

ggsave(paste(out_path, '/cor_N/dian_df15_cor_BAG_Choice_NFLPlasma_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```

###Plasma NFL_log
```{r}
hist(data_car$NFL_Plasma, breaks = 50)
data_car$NFL_log_Plasma = log(data_car$NFL_Plasma)
hist(data_car$NFL_log_Plasma, breaks = 50)
```

```{r}
lme_dbn_NFL_log_Plasma_main = lmer(
  BAG_Choice ~ 
    NFL_log_Plasma
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_log_Plasma_main)


lme_dbn_NFL_log_Plasma_main_anova = anova(lme_dbn_NFL_log_Plasma_main)
F_to_eta2(lme_dbn_NFL_log_Plasma_main_anova$`F value`[1]
          , df = lme_dbn_NFL_log_Plasma_main_anova$NumDF[1]
          , df_error = lme_dbn_NFL_log_Plasma_main_anova$DenDF[1])

lme_dbn_NFL_log_Plasma_interaction = lmer(
  BAG_Choice ~ 
    NFL_log_Plasma*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_NFL_log_Plasma_interaction)

lme_dbn_NFL_log_Plasma_interaction_anova = anova(lme_dbn_NFL_log_Plasma_interaction)
F_to_eta2(lme_dbn_NFL_log_Plasma_interaction_anova$`F value`[6]
          , df = lme_dbn_NFL_log_Plasma_interaction_anova$NumDF[6]
          , df_error = lme_dbn_NFL_log_Plasma_interaction_anova$DenDF[6])

data_car$group_full = data_car$group
levels(data_car$group_full) = c("Full Sample", "Asymptomatic MC", "Symptomatic MC")

cor_NFL_log_plasma = ggplot(data_car, aes(y = BAG_Choice, x = NFL_log_Plasma)) +
  geom_point(aes(color = group_full)) +
  labs(title="Plasma NfL (log)", y = "BAG (Age-Corrected)", x = "Plasma NfL (log)") +
  geom_smooth(aes(color = group_full, fill = group_full), method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group_full)) + 
  scale_color_manual(values = c("black", "orange", "red"), drop = FALSE) +
  scale_fill_manual(values = c("grey", "orange", "red"), drop = FALSE) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  #ylim(min(c(data_car$Age, yscale)), max(c(data_car$Age, yscale))) + 
  stat_cor(aes(color = group_full)#, label.y = c(39.5, 35.5)
           ) +
  theme_bw(base_size = 15) +
  theme(legend.position="bottom", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_NFL_log_plasma

legend_2group <- get_legend(cor_NFL_log_plasma)

ggsave(paste(out_path, '/cor_N/dian_df15_cor_BAG_Choice_NFL_logPlasma_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)
```

###Merge N Fig
```{r}
ggarrange(
  cor_nfl_csf, cor_nfl_plasma

  #, labels = c("A", "B", "C", "D", "E", "F", "G", "H")
  , nrow = 1, ncol = 2
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(out_path, '/poster_figs/dian_df15_BAG_Choice_N_Figure_', ver_date, '.png', sep = ""), width = 10, height = 5, units = "in", dpi = 320)
```

##Cognition
###CDR-SB
```{r}
lme_dbn_CDRSUM.x_main = lmer(
  BAG_Choice ~ 
    CDRSUM.x
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CDRSUM.x_main)

lme_dbn_CDRSUM.x_main_anova = anova(lme_dbn_CDRSUM.x_main)
F_to_eta2(lme_dbn_CDRSUM.x_main_anova$`F value`[1]
          , df = lme_dbn_CDRSUM.x_main_anova$NumDF[1]
          , df_error = lme_dbn_CDRSUM.x_main_anova$DenDF[1])

lme_dbn_CDRSUM.x_interaction = lmer(
  BAG_Choice ~ 
    CDRSUM.x*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_car
)
summary(lme_dbn_CDRSUM.x_interaction)

cor_cdrsum = ggplot(data_car, aes(y = BAG_Choice, x = CDRSUM.x)) +
  geom_point(aes(color = group)) +
  labs(title="CDR-Sum of Boxes", y = "BAG (Age-Corrected)", x = "CDR-SB") +
  geom_smooth(aes(color = group, fill = group), method = lm) +
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group = imagid.x, color = group)) + 
  scale_color_manual(values = c( "orange", "red")) +
  scale_fill_manual(values = c( "orange", "red")) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  stat_cor(aes(color = group), label.y = c(-14, -18), label.x = 9) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.80, .15), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_cdrsum

ggsave(paste(out_path, '/cor_cog/dian_df15_cor_BAG_Choice_CDRSum_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)

```

###Global Composite
```{r}
data_global = subset(data, z_ANIMALS < 6)
data_global = subset(data_global, z_TRAILB < 6)
data_global = subset(data_global, z_MEMUNITS < 6)

lme_dbn_global_cog_main = lmer(
  BAG_Choice ~ 
    global_cog
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_global
)
summary(lme_dbn_global_cog_main)

lme_dbn_global_cog_main_anova = anova(lme_dbn_global_cog_main)
F_to_eta2(lme_dbn_global_cog_main_anova$`F value`[1]
          , df = lme_dbn_global_cog_main_anova$NumDF[1]
          , df_error = lme_dbn_global_cog_main_anova$DenDF[1])

lme_dbn_global_cog_interaction = lmer(
  BAG_Choice ~ 
    global_cog*group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_global
)
summary(lme_dbn_global_cog_interaction)


lme_dbn_global_cog_interaction_anova = anova(lme_dbn_global_cog_interaction)
F_to_eta2(lme_dbn_global_cog_interaction_anova$`F value`[6]
          , df = lme_dbn_global_cog_interaction_anova$NumDF[6]
          , df_error = lme_dbn_global_cog_interaction_anova$DenDF[6])

data_global$group_full = data_global$group
levels(data_global$group_full) = c(levels(data_global$group_full), "Full Sample")

cor_global_cog = ggplot(data_global, aes(y = BAG_Choice, x = global_cog)) +
  geom_point(aes(color = group_full)
    ) +
  labs(title="Global Composite", y = "BAG (Age-Corrected)", x = "Global Composite") +
  geom_smooth(aes(color = group_full, fill = group_full), 
              method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(group_full = imagid.x, color = group_full)
            ) + 
  #scale_fill_discrete(labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample')) +
  scale_color_manual(values = c("blue", "orange", "red", "black")
                     , labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "orange", "red", "grey"), labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = group_full), label.y = c(-13, -16, -19)) +
  theme_bw(base_size = 15) +
  theme(legend.position="bottom", plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_global_cog

legend_3group <- get_legend(cor_global_cog)

ggsave(paste(out_path, '/cor_cog/dian_df15_cor_BAG_Choice_Global_', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)



```

###Global Composite 2 groups
```{r}
data_global = subset(data, z_ANIMALS < 6)
data_global = subset(data_global, z_TRAILB < 6)
data_global = subset(data_global, z_MEMUNITS < 6)

lme_dbn_global_cog_main = lmer(
  BAG_Choice ~ 
    global_cog
  #+ Mutation_Status
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_global
)
summary(lme_dbn_global_cog_main)

lme_dbn_global_cog_main_anova = anova(lme_dbn_global_cog_main)
F_to_eta2(lme_dbn_global_cog_main_anova$`F value`[1]
          , df = lme_dbn_global_cog_main_anova$NumDF[1]
          , df_error = lme_dbn_global_cog_main_anova$DenDF[1])

lme_dbn_global_cog_interaction = lmer(
  BAG_Choice ~ 
    global_cog*Mutation_Status
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  + (1 | fam_id)
  , data = data_global
)
summary(lme_dbn_global_cog_interaction)


lme_dbn_global_cog_interaction_anova = anova(lme_dbn_global_cog_interaction)
F_to_eta2(lme_dbn_global_cog_interaction_anova$`F value`[6]
          , df = lme_dbn_global_cog_interaction_anova$NumDF[6]
          , df_error = lme_dbn_global_cog_interaction_anova$DenDF[6])

data_global$Mutation_Status_full = data_global$Mutation_Status
levels(data_global$Mutation_Status_full) = c(levels(data_global$Mutation_Status_full), "Full Sample")

cor_global_cog_2group = ggplot(data_global, aes(y = BAG_Choice, x = global_cog)) +
  geom_point(aes(color = Mutation_Status_full)
    ) +
  labs(title="Global Cognitive Composite", y = "BAG (Age-Corrected)", x = "Global Cognitive Composite") +
  geom_smooth(aes(color = Mutation_Status_full, fill = Mutation_Status_full), 
              method = lm) + 
  geom_smooth(col = 'black', method = lm, linetype = 'longdash') + 
  geom_line(aes(Mutation_Status_full = imagid.x, color = Mutation_Status_full)
            ) + 
  #scale_fill_discrete(labels = c('Non-Carrier', 'Asymptomatic MC', 'Symptomatic MC', 'Full Sample')) +
  scale_color_manual(values = c("blue",  "red", "black")
                     , labels = c('Non-Carrier', 'Carrier', 'Full Sample'), drop = FALSE
                     ) +
  scale_fill_manual(values = c("blue", "red", "grey"), labels = c('Non-Carrier', 'Carrier', 'Full Sample'), drop = FALSE
                    ) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  #xlim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  #ylim(min(c(data_global$Age, yscale)), max(c(data_global$Age, yscale))) + 
  stat_cor(aes(color = Mutation_Status_full)
           , label.y = c(24, 21)
           , label.x = c(0, 0)
           ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.15, .125), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank())

cor_global_cog_2group

#legend_3Mutation_Status <- get_legend(cor_global_cog)

ggsave(paste(out_path, '/cor_cog/dian_df15_cor_BAG_Choice_Global_2group', ver_date, '.png', sep = ""), width = 6, height = 6, units = "in", dpi = 320)



```

####Merge Cog fig
```{r}
ggarrange(
  cor_global_cog, cor_cdrsum
  , labels = c("A", "B")
  , ncol = 2
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(out_path, '/poster_figs/dian_df15_BAG_Choice_Cognition_Figure_', ver_date, '.png', sep = ""), width = 10, height = 5, units = "in", dpi = 320)
```

###Merge A/T/N/C Fig
```{r}
ggarrange(
  cor_pib, cor_ab4240
  , cor_ptauab40_csf, cor_ptau_plasma
  , cor_nfl_csf, cor_nfl_plasma
  , cor_global_cog, cor_cdrsum
  , labels = c("A", "B", "C", "D", "E", "F", "G", "H")
  , nrow = 4, ncol = 2
  , common.legend = TRUE, legend = "bottom"
  , legend.grob = legend_3group
)

ggsave(paste(out_path, '/poster_figs/dian_df15_BAG_Choice_ATN_Figure_', ver_date, '_lm.png', sep = ""), width = 10, height = 20, units = "in", dpi = 320)

ggsave(paste(ms_path, '/Figure3_ATNC_', ver_date, '.png', sep = ""), width = 10, height = 20, units = "in", dpi = 320)
```

###Merge A/T/N (log) Fig
```{r}
ggarrange(
  cor_PIB_log, cor_ab4240_log
  , cor_ptauab40_log_csf, cor_pTau_log_plasma
  , cor_NFL_log_csf, cor_NFL_log_plasma
  , labels = c("A", "B", "C", "D", "E", "F")
  , nrow = 3, ncol = 2
  , common.legend = TRUE, legend = "bottom"
  , legend.grob = legend_2group
)

ggsave(paste(ms_path, '/Figure3s_ATN_log_', ver_date, '.png', sep = ""), width = 10, height = 15, units = "in", dpi = 320)
```
#Mediation
##PIB_fSUVR_rsf_TOT_CORTMEAN.x

```{r}
lme_cog_PIB_main = lm( #lmer(
  global_cog ~ 
    PIB_fSUVR_rsf_TOT_CORTMEAN.x
  #+ BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_PIB_main)

lme_cog_BAG_PIB_main = lm( #lmer(
  global_cog ~ 
    PIB_fSUVR_rsf_TOT_CORTMEAN.x
  + BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_BAG_PIB_main)

lme_BAG_PIB_main = lm( #lmer(
  BAG_Choice ~ 
    PIB_fSUVR_rsf_TOT_CORTMEAN.x
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_BAG_PIB_main)

set.seed(110)
med_cog_BAG_PIB = mediation::mediate(
  lme_BAG_PIB_main, lme_cog_BAG_PIB_main
  , treat='PIB_fSUVR_rsf_TOT_CORTMEAN.x', mediator='BAG_Choice'
  , boot = TRUE, sims = 1000
  #, robustSE = TRUE
  )

summary(med_cog_BAG_PIB)

png(paste(out_path, '/mediation/A_PIB_', ver_date, '_lm.png', sep = ""), height = 400, width = 400)

medplot_cog_BAG_PIB = plot(med_cog_BAG_PIB, main = "BAG Mediation of Cognition ~ PIB PET")
medplot_cog_BAG_PIB

dev.off()
```
##CSF_AB4240_ratio

```{r}
lme_cog_CSF_AB4240_ratio_main = lm( #lmer(
  global_cog ~ 
    CSF_AB4240_ratio
  #+ BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_CSF_AB4240_ratio_main)

lme_cog_BAG_CSF_AB4240_ratio_main = lm( #lmer(
  global_cog ~ 
    CSF_AB4240_ratio
  + BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_BAG_CSF_AB4240_ratio_main)

lme_BAG_CSF_AB4240_ratio_main = lm( #lmer(
  BAG_Choice ~ 
    CSF_AB4240_ratio
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_BAG_CSF_AB4240_ratio_main)

set.seed(110)
med_cog_BAG_CSF_AB4240_ratio = mediation::mediate(
  lme_BAG_CSF_AB4240_ratio_main, lme_cog_BAG_CSF_AB4240_ratio_main
  , treat='CSF_AB4240_ratio', mediator='BAG_Choice'
  , boot = TRUE, sims = 1000
  #, robustSE = TRUE
  )

summary(med_cog_BAG_CSF_AB4240_ratio)

png(paste(out_path, '/mediation/A_CSF_', ver_date, '_lm.png', sep = ""), height = 400, width = 400)

medplot_cog_BAG_CSF_AB4240_ratio = plot(med_cog_BAG_CSF_AB4240_ratio, main = "BAG Mediation of Cognition ~ CSF AB42/40")
medplot_cog_BAG_CSF_AB4240_ratio

dev.off()
```
##pTau_Plasma

```{r}
lme_cog_pTau_Plasma_main = lm( #lmer(
  global_cog ~ 
    pTau_Plasma
  #+ BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_pTau_Plasma_main)

lme_cog_BAG_pTau_Plasma_main = lm( #lmer(
  global_cog ~ 
    pTau_Plasma
  + BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_BAG_pTau_Plasma_main)

lme_BAG_pTau_Plasma_main = lm( #lmer(
  BAG_Choice ~ 
    pTau_Plasma
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_BAG_pTau_Plasma_main)

set.seed(110)
med_cog_BAG_pTau_Plasma = mediation::mediate(
  lme_BAG_pTau_Plasma_main, lme_cog_BAG_pTau_Plasma_main
  , treat='pTau_Plasma', mediator='BAG_Choice'
  , boot = TRUE, sims = 1000
  #, robustSE = TRUE
  )

summary(med_cog_BAG_pTau_Plasma)

png(paste(out_path, '/mediation/T_Plasma_', ver_date, '_lm.png', sep = ""), height = 400, width = 400)

medplot_cog_BAG_pTau_Plasma = plot(med_cog_BAG_pTau_Plasma, main = "BAG Mediation of Cognition ~ Plasma pTau-181")
medplot_cog_BAG_pTau_Plasma

dev.off()
```

##CSF_ptauab40_ratio

```{r}
lme_cog_CSF_ptauab40_ratio_main = lm( #lmer(
  global_cog ~ 
    CSF_ptauab40_ratio
  #+ BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_CSF_ptauab40_ratio_main)

lme_cog_BAG_CSF_ptauab40_ratio_main = lm( #lmer(
  global_cog ~ 
    CSF_ptauab40_ratio
  + BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_BAG_CSF_ptauab40_ratio_main)

lme_BAG_CSF_ptauab40_ratio_main = lm( #lmer(
  BAG_Choice ~ 
    CSF_ptauab40_ratio
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_BAG_CSF_ptauab40_ratio_main)

set.seed(110)
med_cog_BAG_CSF_ptauab40_ratio = mediation::mediate(
  lme_BAG_CSF_ptauab40_ratio_main, lme_cog_BAG_CSF_ptauab40_ratio_main
  , treat='CSF_ptauab40_ratio', mediator='BAG_Choice'
  , boot = TRUE, sims = 1000
  #, robustSE = TRUE
  )

summary(med_cog_BAG_CSF_ptauab40_ratio)

png(paste(out_path, '/mediation/T_CSF_', ver_date, '_lm.png', sep = ""), height = 400, width = 400)

medplot_cog_BAG_CSF_ptauab40_ratio = plot(med_cog_BAG_CSF_ptauab40_ratio, main = "BAG Mediation of Cognition ~ CSF pTau/40")
medplot_cog_BAG_CSF_ptauab40_ratio

dev.off()
```

##NFL_CSF

```{r}
lme_cog_NFL_CSF_main = lm( #lmer(
  global_cog ~ 
    NFL_CSF
  #+ BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_NFL_CSF_main)

lme_cog_BAG_NFL_CSF_main = lm( #lmer(
  global_cog ~ 
    NFL_CSF
  + BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_BAG_NFL_CSF_main)

lme_BAG_NFL_CSF_main = lm( #lmer(
  BAG_Choice ~ 
    NFL_CSF
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_BAG_NFL_CSF_main)

set.seed(110)
med_cog_BAG_NFL_CSF = mediation::mediate(
  lme_BAG_NFL_CSF_main, lme_cog_BAG_NFL_CSF_main
  , treat='NFL_CSF', mediator='BAG_Choice'
  , boot = TRUE, sims = 1000
  #, robustSE = TRUE
  )

summary(med_cog_BAG_NFL_CSF)

png(paste(out_path, '/mediation/N_CSF_', ver_date, '_lm.png', sep = ""), height = 400, width = 400)

medplot_cog_BAG_NFL_CSF = plot(med_cog_BAG_NFL_CSF, main = "BAG Mediation of Cognition ~ CSF NfL")
medplot_cog_BAG_NFL_CSF

dev.off()
```

##NFL_Plasma

```{r}
lme_cog_NFL_Plasma_main = lm( #lmer(
  global_cog ~ 
    NFL_Plasma
  #+ BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_NFL_Plasma_main)

lme_cog_BAG_NFL_Plasma_main = lm( #lmer(
  global_cog ~ 
    NFL_Plasma
  + BAG_Choice
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_cog_BAG_NFL_Plasma_main)

lme_BAG_NFL_Plasma_main = lm( #lmer(
  BAG_Choice ~ 
    NFL_Plasma
  #+ group
  #+ Age
  #+ codon_cat
  + Sex
  + educ
  + apoe_pos
  #+ (1 | fam_id)
  , data = data_car
)
summary(lme_BAG_NFL_Plasma_main)

set.seed(110)
med_cog_BAG_NFL_Plasma = mediation::mediate(
  lme_BAG_NFL_Plasma_main, lme_cog_BAG_NFL_Plasma_main
  , treat='NFL_Plasma', mediator='BAG_Choice'
  , boot = TRUE, sims = 1000
  #, robustSE = TRUE
  )

summary(med_cog_BAG_NFL_Plasma)

png(paste(out_path, '/mediation/N_Plasma_', ver_date, '_lm.png', sep = ""), height = 400, width = 400)

medplot_cog_BAG_NFL_Plasma = plot(med_cog_BAG_NFL_Plasma, main = "BAG Mediation of Cognition ~ Plasma NfL")
medplot_cog_BAG_NFL_Plasma

dev.off()
```


#Other MRI regional analyses for reference
##HCV
```{r}
data_lme$Mutation_Status = ordered(data_lme$Mutation_Status)

eyo_Hippocampus_mut_lme_all_gam_ci = gamm4( #lmer(
  MR_HCV ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme
  , REML = TRUE
)
summary(eyo_Hippocampus_mut_lme_all_gam_ci$mer)
summary(eyo_Hippocampus_mut_lme_all_gam_ci$gam)

plot(eyo_Hippocampus_mut_lme_all_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_Hippocampus_mut_lme_all_gam_ci$gam)

concurvity(eyo_Hippocampus_mut_lme_all_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-revisited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = unique(Mutation_Status)
  , Sex = "Female"#unique(Sex)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
))

Model = eyo_Hippocampus_mut_lme_all_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_Hippocampus_mut_lme_all_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_Hippocampus_mut_lme_all_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

eyo_Hippocampus_mut_lme_all_gam_ci_pred_car = subset(eyo_Hippocampus_mut_lme_all_gam_ci_pred, Mutation_Status == "Carrier")
eyo_Hippocampus_mut_lme_all_gam_ci_pred_non = subset(eyo_Hippocampus_mut_lme_all_gam_ci_pred, Mutation_Status == "Non-Carrier")

eyo_Hippocampus_mut_lme_all_gam_ci_pred_car$dif1 = eyo_Hippocampus_mut_lme_all_gam_ci_pred_non$lwrS - eyo_Hippocampus_mut_lme_all_gam_ci_pred_car$uprS
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig1 = subset(eyo_Hippocampus_mut_lme_all_gam_ci_pred_car, dif1 > 0)
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig_point1 = min(eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig1$EYO)

eyo_Hippocampus_mut_lme_all_gam_ci_pred_car$dif2 = eyo_Hippocampus_mut_lme_all_gam_ci_pred_non$fit - eyo_Hippocampus_mut_lme_all_gam_ci_pred_car$uprS
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_Hippocampus_mut_lme_all_gam_ci_pred_car, dif2 > 0)
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig2, EYO > -30)
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig_point2 = min(eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig2$EYO)


eyo_Hippocampus_mut_lme_all_gam_ci_pred_car$dif3 = eyo_Hippocampus_mut_lme_all_gam_ci_pred_non$lwrS - eyo_Hippocampus_mut_lme_all_gam_ci_pred_car$fit
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig3 = subset(eyo_Hippocampus_mut_lme_all_gam_ci_pred_car, dif3 > 0)
eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig_point3 = min(eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig3$EYO)

eyo_sig_dif = eyo_Hippocampus_mut_lme_all_gam_ci_pred_sig_point1


cor_eyo_Hippocampus_gam_ci = ggplot(data_lme, aes(x = EYO, y = MR_HCV, colour = Mutation_Status)) +
  geom_point(aes(color = Mutation_Status, shape = Clin_Status)) +
  labs(title="Hippocampus Volume",x="EYO", y = "Volume (mm^3)") +
  geom_smooth(data = eyo_Hippocampus_mut_lme_all_gam_ci_pred
              , aes(color = Mutation_Status
                    #, fill = Mutation_Status
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_Hippocampus_mut_lme_all_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = Mutation_Status
                    , fill = Mutation_Status)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_Hippocampus_mut_lme_all_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = Mutation_Status)) + 
  scale_color_manual(values = c("blue", "red")) +
  scale_fill_manual(values = c("blue", "red")) +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  #geom_hline(yintercept = 0, linetype = "longdash") + 
  geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(round(eyo_sig_dif, 2), 0)
                     , breaks = c(eyo_sig_dif, 0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    

cor_eyo_Hippocampus_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_eyo_Hippocampus_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)
```

##CortSig thickness
```{r}
data_lme$Mutation_Status = ordered(data_lme$Mutation_Status)

eyo_CortSig_mut_lme_all_gam_ci = gamm4( #lmer(
  CortSig_Thickness ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme
  , REML = TRUE
)
summary(eyo_CortSig_mut_lme_all_gam_ci$mer)
summary(eyo_CortSig_mut_lme_all_gam_ci$gam)

plot(eyo_CortSig_mut_lme_all_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_CortSig_mut_lme_all_gam_ci$gam)

concurvity(eyo_CortSig_mut_lme_all_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-revisited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = unique(Mutation_Status)
  , Sex = "Female"#unique(Sex)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
))

Model = eyo_CortSig_mut_lme_all_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_CortSig_mut_lme_all_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_CortSig_mut_lme_all_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

eyo_CortSig_mut_lme_all_gam_ci_pred_car = subset(eyo_CortSig_mut_lme_all_gam_ci_pred, Mutation_Status == "Carrier")
eyo_CortSig_mut_lme_all_gam_ci_pred_non = subset(eyo_CortSig_mut_lme_all_gam_ci_pred, Mutation_Status == "Non-Carrier")

eyo_CortSig_mut_lme_all_gam_ci_pred_car$dif1 = eyo_CortSig_mut_lme_all_gam_ci_pred_non$lwrS - eyo_CortSig_mut_lme_all_gam_ci_pred_car$uprS
eyo_CortSig_mut_lme_all_gam_ci_pred_sig1 = subset(eyo_CortSig_mut_lme_all_gam_ci_pred_car, dif1 > 0)
eyo_CortSig_mut_lme_all_gam_ci_pred_sig_point1 = min(eyo_CortSig_mut_lme_all_gam_ci_pred_sig1$EYO)

eyo_CortSig_mut_lme_all_gam_ci_pred_car$dif2 = eyo_CortSig_mut_lme_all_gam_ci_pred_non$fit - eyo_CortSig_mut_lme_all_gam_ci_pred_car$uprS
eyo_CortSig_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_CortSig_mut_lme_all_gam_ci_pred_car, dif2 > 0)
eyo_CortSig_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_CortSig_mut_lme_all_gam_ci_pred_sig2, EYO > -30)
eyo_CortSig_mut_lme_all_gam_ci_pred_sig_point2 = min(eyo_CortSig_mut_lme_all_gam_ci_pred_sig2$EYO)


eyo_CortSig_mut_lme_all_gam_ci_pred_car$dif3 = eyo_CortSig_mut_lme_all_gam_ci_pred_non$lwrS - eyo_CortSig_mut_lme_all_gam_ci_pred_car$fit
eyo_CortSig_mut_lme_all_gam_ci_pred_sig3 = subset(eyo_CortSig_mut_lme_all_gam_ci_pred_car, dif3 > 0)
eyo_CortSig_mut_lme_all_gam_ci_pred_sig_point3 = min(eyo_CortSig_mut_lme_all_gam_ci_pred_sig3$EYO)

eyo_sig_dif = eyo_CortSig_mut_lme_all_gam_ci_pred_sig_point1

cor_eyo_CortSig_gam_ci = ggplot(data_lme, aes(x = EYO, y = CortSig_Thickness, colour = Mutation_Status)) +
  geom_point(aes(color = Mutation_Status, shape = Clin_Status)) +
  labs(title="Cortical Signature Thickness",x="EYO", y = "Thickness (mm)") +
  geom_smooth(data = eyo_CortSig_mut_lme_all_gam_ci_pred
              , aes(color = Mutation_Status
                    #, fill = Mutation_Status
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_CortSig_mut_lme_all_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = Mutation_Status
                    , fill = Mutation_Status)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_CortSig_mut_lme_all_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = Mutation_Status)) + 
  scale_color_manual(values = c("blue", "red")) +
  scale_fill_manual(values = c("blue", "red")) +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  #geom_hline(yintercept = 0, linetype = "longdash") + 
  geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(round(eyo_sig_dif, 2), 0)
                     , breaks = c(eyo_sig_dif, 0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    

cor_eyo_CortSig_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_eyo_CortSig_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)
```
##Precuneus thickness
```{r}
data_lme$Mutation_Status = ordered(data_lme$Mutation_Status)

eyo_Precuneus_mut_lme_all_gam_ci = gamm4( #lmer(
  MR_LR_PRECUNEUS ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = data_lme
  , REML = TRUE
)
summary(eyo_Precuneus_mut_lme_all_gam_ci$mer)
summary(eyo_Precuneus_mut_lme_all_gam_ci$gam)

plot(eyo_Precuneus_mut_lme_all_gam_ci$gam
     , shade = TRUE
     , seWithMean = TRUE
     , residuals = TRUE
     , pch = 16
     , cex = 0.8
     , all.terms = TRUE, pages = 1
     )

gam.check(eyo_Precuneus_mut_lme_all_gam_ci$gam)

concurvity(eyo_Precuneus_mut_lme_all_gam_ci$gam, full = TRUE)

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-revisited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

Forecast_DF = with(data_lme, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = unique(Mutation_Status)
  , Sex = "Female"#unique(Sex)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
))

Model = eyo_Precuneus_mut_lme_all_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  set.seed(42)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

eyo_Precuneus_mut_lme_all_gam_ci_pred = 
  get_splineBasedCI(Model = eyo_Precuneus_mut_lme_all_gam_ci$gam
                  , Forecast_DF = Forecast_DF
                  )

eyo_Precuneus_mut_lme_all_gam_ci_pred_car = subset(eyo_Precuneus_mut_lme_all_gam_ci_pred, Mutation_Status == "Carrier")
eyo_Precuneus_mut_lme_all_gam_ci_pred_non = subset(eyo_Precuneus_mut_lme_all_gam_ci_pred, Mutation_Status == "Non-Carrier")

eyo_Precuneus_mut_lme_all_gam_ci_pred_car$dif1 = eyo_Precuneus_mut_lme_all_gam_ci_pred_non$lwrS - eyo_Precuneus_mut_lme_all_gam_ci_pred_car$uprS
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig1 = subset(eyo_Precuneus_mut_lme_all_gam_ci_pred_car, dif1 > 0)
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig_point1 = min(eyo_Precuneus_mut_lme_all_gam_ci_pred_sig1$EYO)

eyo_Precuneus_mut_lme_all_gam_ci_pred_car$dif2 = eyo_Precuneus_mut_lme_all_gam_ci_pred_non$fit - eyo_Precuneus_mut_lme_all_gam_ci_pred_car$uprS
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_Precuneus_mut_lme_all_gam_ci_pred_car, dif2 > 0)
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig2 = subset(eyo_Precuneus_mut_lme_all_gam_ci_pred_sig2, EYO > -30)
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig_point2 = min(eyo_Precuneus_mut_lme_all_gam_ci_pred_sig2$EYO)


eyo_Precuneus_mut_lme_all_gam_ci_pred_car$dif3 = eyo_Precuneus_mut_lme_all_gam_ci_pred_non$lwrS - eyo_Precuneus_mut_lme_all_gam_ci_pred_car$fit
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig3 = subset(eyo_Precuneus_mut_lme_all_gam_ci_pred_car, dif3 > 0)
eyo_Precuneus_mut_lme_all_gam_ci_pred_sig_point3 = min(eyo_Precuneus_mut_lme_all_gam_ci_pred_sig3$EYO)

eyo_sig_dif = eyo_Precuneus_mut_lme_all_gam_ci_pred_sig_point1

cor_eyo_Precuneus_gam_ci = ggplot(data_lme, aes(x = EYO, y = MR_LR_PRECUNEUS, colour = Mutation_Status)) +
  geom_point(aes(color = Mutation_Status, shape = Clin_Status)) +
  labs(title="Precuneus Thickness",x="EYO", y = "Thickness (mm)") +
  geom_smooth(data = eyo_Precuneus_mut_lme_all_gam_ci_pred
              , aes(color = Mutation_Status
                    #, fill = Mutation_Status
                    , x = EYO, y = fit)
              ) + 
  geom_ribbon(data = eyo_Precuneus_mut_lme_all_gam_ci_pred
              , aes(ymin = lwrS, ymax = uprS
                    , x = EYO, y = fit
                    #, color = Mutation_Status
                    , fill = Mutation_Status)
              , alpha = 0.4
              , colour = NA
              ) +
  #geom_line(aes(y = predict(eyo_Precuneus_mut_lme_all_spline
                            
                            #, re.form = ~(1 | fam_id)
  #                          ))) +
  #geom_line(aes(group = imagid.x, color = Mutation_Status)) + 
  scale_color_manual(values = c("blue", "red")) +
  scale_fill_manual(values = c("blue", "red")) +
  geom_vline(xintercept = 0, linetype = "longdash") + 
  #geom_hline(yintercept = 0, linetype = "longdash") + 
  geom_vline(xintercept = eyo_sig_dif, linetype = "solid") +
  #annotate("text", x = eyo_sig_dif
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(eyo_sig_dif, 2))
  #         , parse = TRUE) +
  #annotate("text", x = 0
  #         , y = min(data_lme$BAG_Choice)
  #         , label = paste("", round(0, 2))
  #         , parse = TRUE) +


  #xlim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #ylim(min(c(data$Age, yscale)), max(c(data$Age, yscale))) + 
  #xlim(-35, 15) +
  scale_x_continuous(labels = c(round(eyo_sig_dif, 2), 0)
                     , breaks = c(eyo_sig_dif, 0)
                     , limits = c(-35, 15)
                     ) +
  theme_bw(base_size = 15) +
  theme(legend.position=c(.10, .825), plot.title = element_text(hjust = 0.5), legend.title = element_blank(), legend.key = element_blank(), legend.background=element_blank()
        #, axis.text.x = element_blank(), axis.ticks.x = element_blank()
        , panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()
        )
    

cor_eyo_Precuneus_gam_ci

ggsave(paste(out_path, '/cor_eyo/dian_df15_cor_eyo_Precuneus_gam_ci_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)
```

##Bootstrap GAM
```{r}
n_sims = 10000
eyo_sig_dif_sims = data.frame("sim" = 1:n_sims)
eyo_sig_dif_sims$BAG = NA
eyo_sig_dif_sims$Hippocampus = NA
eyo_sig_dif_sims$CortSig = NA
eyo_sig_dif_sims$Precuneus = NA

data_lme$Mutation_Status = ordered(data_lme$Mutation_Status)

#wrapper function to run gamm4 model

gamm4_eyo <- function(sample, dv){
  sample = as.data.frame(sample)
  dv_vec = as.numeric(sample[[dv]])
  
  model = gamm4( #lmer(
    dv_vec ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
  + Sex
  + educ
  + apoe_pos
  , random = ~ (1 | fam_id)
  , data = sample
  , REML = TRUE
  )
  return(model)
}

#gamm4_eyo(sample = data_sim, dv = 'BAG_Choice')

#from https://fromthebottomoftheheap.net/2016/12/15/simultaneous-interval-revisited/
rmvn <- function(n, mu, sig) { ## MVN random deviates
  L <- mroot(sig)
  m <- ncol(L)
  t(mu + L %*% matrix(rnorm(m*n), m, n))
}

generate_forecast <- function(sample){
  Forecast_DF = with(sample, expand.grid(
  EYO = unique(EYO)
  , Mutation_Status = unique(Mutation_Status)
  , Sex = "Female"#unique(Sex)
  , educ = mean(data$educ, na.rm = T)#unique(educ)
  , apoe_pos = "E4-"#unique(apoe_pos)
  #, fam_id = unique(fam_id)
  ))
  return(Forecast_DF)
}

#generate_forecast(data_lme)

Model = eyo_Precuneus_mut_lme_all_gam_ci$gam

#from Julie
get_splineBasedCI <- function(Model, Forecast_DF){
  Vb <- vcov(Model)
  pred <- predict(Model, Forecast_DF, se.fit = TRUE)
  se.fit <- pred$se.fit
  #set.seed(sim)
  N <- 10000
 
  BUdiff <- rmvn(N, mu = rep(0, nrow(Vb)), sig = Vb)
  Cg <- predict(Model, Forecast_DF, type = "lpmatrix")
  simDev <- Cg %*% t(BUdiff)
 
  absDev <- abs(sweep(simDev, 1, se.fit, FUN = "/"))
  masd <- apply(absDev, 2L, max)
  crit <- quantile(masd, prob = 0.834, type = 8)
 
  pred <- transform(cbind(data.frame(pred), Forecast_DF),
                    uprP = fit + (2 * se.fit),
                    lwrP = fit - (2 * se.fit),
                    uprS = fit + (crit * se.fit),
                    lwrS = fit - (crit * se.fit))
  
  #sims <- rmvn(N, mu = coef(Model), sig = Vb)
  #fits <- Cg %*% t(sims)
  
  #nrnd <- 30
  #rnd <- sample(N, nrnd)
  #stackFits <- stack(as.data.frame(fits[, rnd]))
  #stackFits <- transform(stackFits, EYO = rep(newd$EYO, length(rnd)))

  return(pred)
}

get_eyo_dif_bag <- function(ci){
  ci_car = subset(ci, Mutation_Status == "Carrier")
  ci_non = subset(ci, Mutation_Status == "Non-Carrier")

  ci_car$dif1 = ci_car$lwrS - ci_non$uprS
  ci_sig1 = subset(ci_car, dif1 > 0)
  ci_sig1 = subset(ci_sig1, EYO > -20)
  ci_sig_point1 = min(ci_sig1$EYO)

  ci_car$dif2 = ci_car$fit - ci_non$uprS
  ci_sig2 = subset(ci_car, dif2 > 0)
  ci_sig2 = subset(ci_sig2, EYO > -20)
  ci_sig_point2 = min(ci_sig2$EYO)

  ci_car$dif3 = ci_car$lwrS - ci_non$fit
  ci_sig3 = subset(ci_car, dif3 > 0)
  ci_sig_point3 = min(ci_sig3$EYO)

  eyo_sig_dif = ci_sig_point1 #ci_sig_point2
  return(eyo_sig_dif)
}

get_eyo_dif_mr <- function(ci){
  ci_car = subset(ci, Mutation_Status == "Carrier")
  ci_non = subset(ci, Mutation_Status == "Non-Carrier")

  ci_car$dif1 = ci_non$lwrS - ci_car$uprS
  ci_sig1 = subset(ci_car, dif1 > 0)
  ci_sig1 = subset(ci_sig1, EYO > -20)
  ci_sig_point1 = min(ci_sig1$EYO)

  ci_car$dif2 = ci_non$fit - ci_car$uprS
  ci_sig2 = subset(ci_car, dif2 > 0)
  ci_sig2 = subset(ci_sig2, EYO > -20)
  ci_sig_point2 = min(ci_sig2$EYO)

  ci_car$dif3 = ci_non$lwrS - ci_car$fit
  ci_sig3 = subset(ci_car, dif3 > 0)
  ci_sig3 = subset(ci_sig3, EYO > -20)
  ci_sig_point3 = min(ci_sig3$EYO)

  eyo_sig_dif = ci_sig_point1 #ci_sig_point3
  return(eyo_sig_dif)
}

set.seed(1)
for(sim in 1:n_sims){#sim = 1
  
  sim_idxs_car = sample(1:length(data_lme_car$newid16), length(data_lme_car$newid16), replace = TRUE)
  data_sim_car = data_lme_car[sim_idxs_car, ]
  data_sim_car = data_sim_car[order(data_sim_car$EYO),]
  
  sim_idxs_non = sample(1:length(data_lme_non$newid16), length(data_lme_non$newid16), replace = TRUE)
  data_sim_non = data_lme_non[sim_idxs_non, ]
  data_sim_non = data_sim_non[order(data_sim_non$EYO),]
  
  data_sim = rbind(data_sim_car, data_sim_non)
  
  data_sim$Mutation_Status = ordered(data_sim$Mutation_Status)
  
  newdata_sim = generate_forecast(data_sim)
  
  #sample = data_sim
  #model_sim_bag = gamm4_eyo(data_sim, 'BAG_Choice')$gam
  
  #BAG
  model_sim_bag = gamm4(BAG_Choice ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
    + Sex
    + educ
    + apoe_pos
  , random = ~ (1 | fam_id), data = data_sim, REML = TRUE)$gam
  
  #print(summary(model_sim_bag))
  
  #plot(model_sim_bag, shade = TRUE, seWithMean = TRUE, residuals = TRUE, pch = 16, cex = 0.8)

  ci_sim_bag = get_splineBasedCI(Model = model_sim_bag
                                 , Forecast_DF = newdata_sim)
  
  eyo_sig_dif_sims$BAG[sim] = get_eyo_dif_bag(ci_sim_bag)
  
  #Hippocampus
  model_sim_Hippocampus = gamm4(MR_HCV ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
    + Sex
    + educ
    + apoe_pos
  , random = ~ (1 | fam_id), data = data_sim, REML = TRUE)$gam
  
  ci_sim_Hippocampus = get_splineBasedCI(Model = model_sim_Hippocampus
                                 , Forecast_DF = newdata_sim)
  
  eyo_sig_dif_sims$Hippocampus[sim] = get_eyo_dif_mr(ci_sim_Hippocampus)
  
  #CortSig
  model_sim_CortSig = gamm4(CortSig_Thickness ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
    + Sex
    + educ
    + apoe_pos
  , random = ~ (1 | fam_id), data = data_sim, REML = TRUE)$gam
  
  ci_sim_CortSig = get_splineBasedCI(Model = model_sim_CortSig
                                 , Forecast_DF = newdata_sim)
  
  eyo_sig_dif_sims$CortSig[sim] = get_eyo_dif_mr(ci_sim_CortSig)
  
  #Precuneus
  model_sim_Precuneus = gamm4(MR_LR_PRECUNEUS ~ 
    Mutation_Status
    + s(EYO, k = 4, bs = 'cr')
    + s(EYO
        , k = 4, bs = 'cr'
        , by = Mutation_Status
        )
    + Sex
    + educ
    + apoe_pos
  , random = ~ (1 | fam_id), data = data_sim, REML = TRUE)$gam
  
  ci_sim_Precuneus = get_splineBasedCI(Model = model_sim_Precuneus
                                 , Forecast_DF = newdata_sim)
  
  eyo_sig_dif_sims$Precuneus[sim] = get_eyo_dif_mr(ci_sim_Precuneus)

}

eyo_ci_BAG = quantile(eyo_sig_dif_sims$BAG, c(.025, .5, .975))
eyo_ci_Hippocampus = quantile(eyo_sig_dif_sims$Hippocampus, c(.025, .5, .975))
eyo_ci_CortSig = quantile(eyo_sig_dif_sims$CortSig, c(.025, .5, .975))
eyo_ci_Precuneus = quantile(eyo_sig_dif_sims$Precuneus, c(.025, .5, .975))

eyo_sig_dif_sims$Hippocampus_dif = eyo_sig_dif_sims$Hippocampus - eyo_sig_dif_sims$BAG
eyo_p_BAGvHippocampus = length(subset(eyo_sig_dif_sims, Hippocampus_dif <= 0)$Hippocampus_dif) / length(eyo_sig_dif_sims$Hippocampus_dif)

eyo_sig_dif_sims$CortSig_dif = eyo_sig_dif_sims$CortSig - eyo_sig_dif_sims$BAG
eyo_p_BAGvCortSig = length(subset(eyo_sig_dif_sims, CortSig_dif <= 0)$CortSig_dif) / length(eyo_sig_dif_sims$CortSig_dif)

eyo_sig_dif_sims$Precuneus_dif = eyo_sig_dif_sims$Precuneus - eyo_sig_dif_sims$BAG
eyo_p_BAGvPrecuneus = length(subset(eyo_sig_dif_sims, Precuneus_dif <= 0)$Precuneus_dif) / length(eyo_sig_dif_sims$Precuneus_dif)

write.csv(eyo_sig_dif_sims, paste(out_path, '/cor_eyo/eyo_sig_dif_sims_', ver_date, '.csv', sep = ""))

#eyo_sig_dif_sims = read.csv(paste(out_path, '/cor_eyo/eyo_sig_dif_sims_231011.csv', sep = ""))

eyo_sig_dif_sims_long = gather(eyo_sig_dif_sims, Measurement, EYO, BAG:Precuneus)
eyo_sig_dif_sims_long$Measurement = recode_factor(eyo_sig_dif_sims_long$Measurement
                                                  , "Hippocampus" = "Hippocampus Volume"
                                                  , "Precuneus" = "Precuneus Thickness"
                                                  , "CortSig" = "Cortical Signature Thickness"
                                                  , "BAG" = "BAG"
                                                  )

hist(eyo_sig_dif_sims_long$EYO, breaks = 25)

violin_eyo_sig_dif_sims = ggplot(eyo_sig_dif_sims_long, aes(y=Measurement, x=EYO, fill=Measurement)) +
  labs(title="Comparison of MRI Measures", y="", x = "EYO of ADAD Divergence") +
  geom_violin(trim=FALSE, color = "black") + 
  #scale_fill_manual(breaks=c("Non-Carrier", "Asymptomatic MC", "Symptomatic MC"),
  #                            values=c("blue", "orange", "red")) +
  #scale_fill_viridis(discrete = TRUE) + 
  scale_fill_tableau()+#palette = "Tableau 10") +

  geom_vline(xintercept = 0, linetype = "longdash") +
  geom_boxplot(width=0.3, color = "black") +
  #geom_jitter(shape=16, position=position_jitter(0.05)) +
  #stat_compare_means(comparisons = my_comparisons, method = "wilcox.test"
  #                   , aes(label = ..p.adj..)
  #                   )+ # Add pairwise comparisons p-value
  #stat_compare_means(label.y = 50)+     # Add global p-value
  theme_classic(base_size = 15) +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5))

violin_eyo_sig_dif_sims

ggsave(paste(out_path, '/cor_eyo/violin_eyo_sig_dif_sims_', ver_date, '.png', sep = ""), width = 10, height = 6, units = "in", dpi = 320)

```
###EndBootstrap

##Merge MRI Comparison Figure
```{r}
ggarrange(
  cor_eyo_CortSig_gam_ci, cor_eyo_Precuneus_gam_ci
  , cor_eyo_Hippocampus_gam_ci, violin_eyo_sig_dif_sims
  , nrow = 2, ncol = 2
  , labels = c("A", "B", "C", "D")
  , common.legend = TRUE, legend = "bottom"
)
ggsave(paste(ms_path, '/Figure4_MRIComparison_', ver_date, '.png', sep = ""), width = 12, height = 8, units = "in", dpi = 320)

```


#END

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Ctrl+Alt+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Ctrl+Shift+K* to preview the HTML file).

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.
